---
title: "snRNA-seq cell type proportions, IHC plots, covariate plots, quantitative covariate analysis"
output:
  html_notebook:
    toc: TRUE
    toc_float: TRUE
    theme: united
author: "MB"
---



# Load libraries

```{r message=FALSE, warning=FALSE}
  library(pagoda2)
  library(conos)
  library(magrittr)
  library(scrattch.io)
  library(readr)
  library(data.table)
  library(ggplot2)
  library(gridExtra)
  library(pals)
  library(uwot)
  library(forcats)
  library(ggforce)
  library(Hmisc)
  library(viridis)
  library(rhdf5) #from bioconductor, for hdf5 matrices
  library(readxl) #import xlsx files
  library(dplyr)
  library(writexl)
  library(cowplot) #plot_grid function
  library(patchwork)
  library(ggalluvial)
  library("ggrepel")
  library("car") #leveneTest
  library("ggsignif") #geom_signif stat_signif
  library("ggrastr") #for rasterization of plots!!!!
  library("DESeq2")
  library(Seurat)
  library("parallel")
  library(reshape2) #for melt to reshate data frames
  library(scales) #for different formating of scale numbers
  library("ggrastr")
  
  
  
```


# Import Schizo data, annotations, palettes

```{r}
con_scz <- Conos$new(read_rds("~/projects/human_schizo/MB6-MB23/2ndary/con.rds"))

#con_scz <- Conos$new(con_scz)

#Annotations
annotations_scz2 <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/annotations_scz2.rds")

#Subtype orders
msub_order <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/msub_order.rds")
hsub_order <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/hsub_order.rds")

#Palettes

palette_45 <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/palette_45.rds")
palette_45_2 <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/palette_45_2.rds")

high.pal <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/high.pal.rds")

#write_rds(high.pal, "rds_objects/high.pal.rds")

med.pal <- setNames(c("#5A5156", "#F6222E", "#FEAF16", "#B00068", "#2ED9FF", "#AA0DFE", "#F8A19F", "#325A9B", "#85660D", 
                    "#B10DA1", "#FC1CBF", "#C075A6", "#782AB6", "#1C7F93", "#683B79", "#993F00"), levels(annotations_scz2$med))


#write_rds(med.pal, "rds_objects/med.pal.rds")


set.seed(0)
sample.pal <- setNames(sample(rainbow(length(levels(con_scz$getDatasetPerCell())))),levels(con_scz$getDatasetPerCell()))







#load Allen annotation
annotation_allen <- fread(file = "~/projects/human_schizo/MB6-MB23/2ndary/Allen_31.01.2020/sample_annotations.csv")



#scurblet scores for doublet filtration

scrubletf <- read_rds("~/projects/human_schizo/MB6-MB23/2ndary/scrubletf.rds")




```



# No MB8 annotation - for plotting, DE etc

```{r}

annotations_scz2_no_MB8 <- lapply(annotations_scz2, function(x) {
  
  x <- x[! grepl("MB8_", names(x))]
  x
  
  
})





```



# Variables

```{r}
samplegroups <- list(
  control = c('MB7', 'MB9', 'MB11', 'MB13', 'MB15', 'MB16', 'MB17', 'MB19', 'MB20', 'MB21', 'MB18-2'),
  schizo = c('MB6', 'MB10', 'MB12', 'MB14', 'MB22', 'MB23','MB8-2', 'MB8')
)


diseasef <- as.factor(setNames(rep(names(samplegroups),unlist(lapply(samplegroups,length))),unlist(samplegroups)))

celldiseasef <- setNames(diseasef[as.vector(con_scz$getDatasetPerCell())], names(con_scz$getDatasetPerCell()))






```



# Variables no MB8

```{r}
samplegroups_no_MB8 <- list(
  control = c('MB7', 'MB9', 'MB11', 'MB13', 'MB15', 'MB16', 'MB17', 'MB19', 'MB20', 'MB21', 'MB18-2'),
  schizo = c('MB6', 'MB10', 'MB12', 'MB14', 'MB22', 'MB23','MB8-2')
)


diseasef_no_MB8 <- as.factor(setNames(rep(names(samplegroups_no_MB8),
                                          unlist(lapply(samplegroups_no_MB8,length))),unlist(samplegroups_no_MB8)))





```





# Batches

```{r}
#Import Metadata

#on 2.10.2020 MB6 cause of death and histopathology were added

metadata <- read.csv(
  "~/projects/human_schizo/MB6-MB23/2ndary/Scz_Experiments_May_June_2019_MB6_MB23_METADATA_updated_2.10.2020.csv", 
  header = T)


metadata$Diagnosis %<>% gsub("Ctr", "control", .) %>% gsub("Scz", "schizo", .)

```






#Variables extra

```{r}
#random palette
rpal <- function(n) {sample(rainbow(n))}



#pH
phfactor <- setNames(metadata$pH_liquor,
                     metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#colors pH
color.ph <-
       colorRamps::matlab.like2(n = 
                                  (length((min(metadata$pH_liquor %>% as.vector %>% `[`(. != "n.a.") %>% as.numeric) * 100) : 
                                    (max(metadata$pH_liquor %>% as.vector %>% `[`(. != "n.a.") %>% as.numeric) * 100)))) %>%
  
  
      setNames((min(metadata$pH_liquor %>% as.vector %>% `[`(. != "n.a.") %>% as.numeric) * 100) : 
                 (max(metadata$pH_liquor %>% as.vector %>% `[`(. != "n.a.") %>% as.numeric) * 100)) %>%
      `[`(., ((metadata$pH_liquor %>% as.vector %>% `[`(. != "n.a.") %>% as.numeric) * 100) %>% 
            unique %>% sort %>% as.character) %>% as.vector %>% rev %>% c(., "grey92")


#Sex
sexfactor <- setNames(metadata$Gender,
                      metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#Age
agefactor <- setNames(metadata$Age,
                      metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#colors Age
color.age <-
    colorRamps::matlab.like2(n = length(min(metadata$Age) : max(metadata$Age))) %>%
      setNames(min(metadata$Age) : max(metadata$Age)) %>%
      `[`(., metadata$Age %>% unique %>% sort %>% as.character) %>% as.vector


#PMI
pmifactor <- setNames(metadata$PMI_hrs,
                      metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#colors PMI

color.PMI <-
    colorRamps::matlab.like2(n = (length((min(metadata$PMI_hrs) * 100) : (max(metadata$PMI_hrs) * 100)))) %>%
      setNames((min(metadata$PMI_hrs) * 100) : (max(metadata$PMI_hrs) * 100)) %>%
      `[`(., (metadata$PMI_hrs * 100) %>% unique %>% sort %>% as.character) %>% as.vector



#Source
sourcefactor <- setNames(metadata$Source,
                         metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


# Cause of death
deathfactor <- 
  setNames(metadata$Cause_of_death %>% factor(., levels = c(`[`(unique(.), unique(.) != "n.a."), "n.a.")),
           metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#Colors cause of death
color.death <- c(cols25()[1:length(metadata$Cause_of_death %>% unique) - 1], "grey92")


#FACS
facsfactor <- setNames(metadata$FACS_10x_1st_day_date,
                       metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#% NeuN+
neunfactor <- setNames(metadata$pcent_NeuN_positive,
                       metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#colors NeuN
color.neun <-
      colorRamps::matlab.like2(n = (length((min(metadata$pcent_NeuN_positive) * 10) : 
                                             (max(metadata$pcent_NeuN_positive) * 10)))) %>%
      setNames((min(metadata$pcent_NeuN_positive) * 10) : (max(metadata$pcent_NeuN_positive) * 10)) %>%
      `[`(., (metadata$pcent_NeuN_positive * 10) %>% unique %>% sort %>% as.character) %>% as.vector %>% rev



#GEM kit lot
gemfactor <- setNames(metadata$X10x_GEM_kit_lot,
                      metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#cDNA_clean_pre-amp date

preampfactor <- setNames(metadata$cDNA_cleanup_preAmp,
                         metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#library prep date
libdatefactor <- setNames(metadata$library_prep_date,
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Lib prep kit lot
libfactor <- setNames(metadata$X10_libr_prep_kit_batch,
                      metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#volume nucl susp loaded
volfactor <- setNames(metadata$Volume_susp_loaded,
                      metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#Color volume susp loaded
color.vol <-
      colorRamps::matlab.like2(n = (length((min(metadata$Volume_susp_loaded) * 10) : 
                                             (max(metadata$Volume_susp_loaded) * 10)))) %>%
      setNames((min(metadata$Volume_susp_loaded) * 10) : (max(metadata$Volume_susp_loaded) * 10)) %>%
      `[`(., (metadata$Volume_susp_loaded * 10) %>% unique %>% sort %>% as.character) %>% as.vector



#cDNA concentration

cdnafactor <- setNames(metadata$cDNA_concentration,
                       metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#Color cDNA concentration
color.cdna <-
      colorRamps::matlab.like2(n = (length((min(metadata$cDNA_concentration) * 100) : 
                                             (max(metadata$cDNA_concentration) * 100)))) %>%
      setNames((min(metadata$cDNA_concentration) * 100) : (max(metadata$cDNA_concentration) * 100)) %>%
      `[`(., (metadata$cDNA_concentration * 100) %>% unique %>% sort %>% as.character) %>% as.vector %>% rev


#ng per library prep
amountlibfactor <- setNames(metadata$Amount_.ng._per_reaction,
                            metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Index primer
indexfactor <- setNames(metadata$Index_primer,
                        metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Autopsy brain hypoxia signs
hypoxiafactor <- setNames(factor(metadata$Postmortem_signs_of_brain_hipoxia %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .),
                                 levels = c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Colors Y/N + n.a.

color.yn <- c('dodgerblue1', 'indianred1', "grey92")


#Tangles
tanglesfactor <- setNames(factor(metadata$Dementia_tangles_N.no._Low_Mid_High_Very_High %>%
                                   gsub("N", "No", .) %>% gsub("Mid", "Med", .), 
                                 
                                 levels = c("No", "Low", "Med", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#color tangles
color.tangles <- c(colorRamps::matlab.like2(n = (metadata$Dementia_tangles_N.no._Low_Mid_High_Very_High %>%
                                   gsub("N", "No", .) %>% gsub("Mid", "Med", .) %>% 
                                                   unique %>% length) -1), "grey92")

#Amyloid
amyloidfactor <- setNames(factor(metadata$Dementia_amyloid_N.no._Low_Mid_High_Very_High %>%
                                   gsub("N", "No", .) %>% gsub("Mid", "Med", .), 
                                 levels = c("No", "Low", "Med", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Color Amyloid

color.amyloid <- c(colorRamps::matlab.like2(n = (metadata$Dementia_amyloid_N.no._Low_Mid_High_Very_High %>%
                                   gsub("N", "No", .) %>% gsub("Mid", "Med", .) %>% 
                                                   unique %>% length) -1), "grey92")


#Atherosclerosis
atherofactor <- setNames(factor(metadata$Brain_vasculature_atherosclerosis_N.No._Low_Mid_High %>%
                                   gsub("N", "No", .) %>% gsub("Mid", "Med", .), 
                                levels = 
                                  c("No", "Low", "Med", "High", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Color Amyloid

color.athero <- c(colorRamps::matlab.like2(n = (metadata$Brain_vasculature_atherosclerosis_N.No._Low_Mid_High %>%
                                   gsub("N", "No", .) %>% gsub("Mid", "Med", .) %>% 
                                                   unique %>% length) -1), "grey92")


#Lewy bodies
lewyfactor <- setNames(factor(metadata$Parkinsonism_Lewy_bodies_N.no._Low_Mid_High_Very_High %>%
                                   gsub("N", "No", .), 
                              levels = 
                                  c("No", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Color lewy

color.lewy <- c("dodgerblue1", "grey92")


#Lithium treatment
lithiumfactor <- setNames(factor(metadata$Lithium %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#neuroleptic duration
neuroleptfactor <- setNames(factor(metadata$Neuroleptic_treatment_duration_years, levels = 
                                     c(0, 0.003, 0.16, 2, 10, 30, 40, "n.a.")),
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#Color neuroleptic duration years
color.neurolept <-
       c(colorRamps::matlab.like2(n = (metadata$Neuroleptic_treatment_duration_years %>% 
                                                   unique %>% length) -1), "grey92")


#typical neuroleptics
typicalfactor <- setNames(factor(metadata$Typical_neuroleptics %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#atypical neuroleptics
atypicalfactor <- setNames(factor(metadata$Atypical_neuroleptics %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


 
#tranquilizers benzodiazepines
benzofactor <- setNames(factor(metadata$Tranquilizers_benzodiazepines %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#tranquilizers barbiturates
barbifactor <- setNames(factor(metadata$Tranquilizers_barbiturates %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Antidepressants tricyclic
tricycfactor <- setNames(factor(metadata$Antidepressant_tricyclic %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))




#Antidepressants SIRT/NE inhib
sirtfactor <- setNames(factor(metadata$Antidepressant_serotonin_norepinephrine.transporter_inhibitors %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))




#Chemo tamoxifen
tamoxifactor <- setNames(factor(metadata$Chemotherapy_tamoxifen %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Chemo antiprolif
proliffactor <- setNames(factor(metadata$Chemotherapy_anti_proliferative %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Radiotherapy
radiofactor <- setNames(factor(metadata$Radiotherapy %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Cancer
cancerfactor <- setNames(factor(metadata$Cancer %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#Onset of psychiatry

onsetfactor <- setNames(factor(metadata$First_psychiatric_symptoms_age, 
                               levels = metadata$First_psychiatric_symptoms_age %>% 
                                 as.vector %>% `[`(. != "n.a.") %>% `[`(. != "No") %>% as.numeric %>%
                                           unique %>% sort %>% c(., "No", "n.a.")),
                               metadata$Identifier)[
                                 as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


color.onset <-
      colorRamps::matlab.like2(n = 
                                  (length((min(metadata$First_psychiatric_symptoms_age %>% as.vector %>% 
                                                 `[`(. != "n.a.")  %>% `[`(. != "No") %>% as.numeric)) : 
                                    (max(metadata$First_psychiatric_symptoms_age %>% as.vector %>% 
                                           `[`(. != "n.a.")  %>% `[`(. != "No") %>% as.numeric))))) %>%
  
  
      setNames((min(metadata$First_psychiatric_symptoms_age %>% as.vector %>% `[`(. != "n.a.") %>% `[`(. != "No") %>%
                      as.numeric)) : 
                 (max(metadata$First_psychiatric_symptoms_age %>% as.vector %>% `[`(. != "n.a.") %>% `[`(. != "No")  %>%
                        as.numeric))) %>%
      `[`(., ((metadata$First_psychiatric_symptoms_age %>% as.vector %>% `[`(. != "n.a.") %>% `[`(. != "No") %>%
                 as.numeric)) %>% 
            unique %>% sort %>% as.character) %>% as.vector %>% c(., "thistle2", "grey92")



#Hyperthension
hyperfactor <- setNames(factor(metadata$Hypertension %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Depression
deprfactor <- setNames(factor(metadata$Depression %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Malnutrition/weight loss
malnutrfactor <- setNames(factor(metadata$Malnutrition_weight_loss %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))




#breathing problems
breathfactor <- setNames(factor(metadata$Breathing_problems %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Opioid
opioidfactor <- setNames(factor(metadata$Opioid_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Dopamine
dopaminefactor <- setNames(factor(metadata$Dopaminergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Serotonine
serotoninefactor <- setNames(factor(metadata$Serotoninergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Glu-ergic
glufactor <- setNames(factor(metadata$Glutamatergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#GABA
gabafactor <- setNames(factor(metadata$GABAergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Histamine
histfactor <- setNames(factor(metadata$Histaminergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Muscarine
muscarfactor <- setNames(factor(metadata$Muscarinic_acetycholinergic__modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))




#Nicotinic
nicofactor <- setNames(factor(metadata$Nicotinic_acetylcholine_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Adrenergic
adrenfactor <- setNames(factor(metadata$Adrenergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Adenosin
adenfactor <- setNames(factor(metadata$Adenosinergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Glycine
glycinefactor <- setNames(factor(metadata$Glycinergic_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))



#Na
nafactor <- setNames(factor(metadata$Na_channels_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))

#K
kfactor <- setNames(factor(metadata$K_channels_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Ca
cafactor <- setNames(factor(metadata$Ca_channels_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Proton pump inhibit
protonfactor <- setNames(factor(metadata$Proton_pump_inhibitors_PPI %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#MAO inhibitors
maofactor <- setNames(factor(metadata$MAO_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Mitoch electron transport
mitofactor <- setNames(factor(metadata$Mitochondrial_electron_transport_modulators %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))


#Smoking
smokefactor <- setNames(factor(metadata$Nicotine_abuse %>%
                                   gsub("N", "No", .) %>% gsub("Y", "Yes", .), 
                                 levels = 
                                  c("No", "Yes", "n.a.")), 
                          metadata$Identifier)[as.vector(con_scz$getDatasetPerCell()[names(annotations_scz2$high)])] %>%
  setNames(names(con_scz$getDatasetPerCell()[names(annotations_scz2$high)]))




```













## Plot Annotations High
```{r fig.height=4.2, fig.width=4.2}




plot.anno.high <- 
      con_scz$plotGraph(groups = annotations_scz2$high,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, 
        raster = T, raster.dpi = 600, 
       # raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Subtypes, high resolution") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none") +
  scale_color_manual(values = high.pal)


plot.anno.high

```



## Plot Samples
```{r fig.height=4.2, fig.width=6}




plot.batch.samples <- 
      con_scz$plotGraph(groups = con_scz$getDatasetPerCell()[
        names(con_scz$getDatasetPerCell()) %in% names(annotations_scz2$high)],
                    alpha = 0.03, size = 0.1, palette = cols25(),
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, 
        raster = T, raster.dpi = 600,
     #   raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Samples") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2))


plot.batch.samples

```



## Plot pH
```{r fig.height=4.2, fig.width=5}
plot.batch.ph <- 
      con_scz$plotGraph(groups = phfactor,
                    alpha = 0.05, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Liquor pH") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
 scale_color_manual(values = color.ph)
#  scale_color_manual(values = c("#9E0142", "#D53E4F", "#F46D43", "#FDAE61", "#FEE08B", "yellow", "#E6F598", "#ABDDA4",
#                                "#66C2A5", "#3288BD", "#5E4FA2", "grey92"))


plot.batch.ph
```


## Plot Disease-Control

```{r fig.height=4.2, fig.width=5}
plot.batch.disease <- 
      con_scz$plotGraph(groups = celldiseasef,
                    alpha = 0.015, size = 0.01,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, 
        raster = T, raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Condition") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +

  scale_color_manual(values = palette_45_2[c(8, 15)]
                       
                       #c('dodgerblue1', 'indianred1')
                       
                       )


plot.batch.disease


```

## Plot Sex
```{r  fig.height=4.2, fig.width=4.2}
plot.batch.sex <- 
      con_scz$plotGraph(groups = sexfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, 
        raster = T, raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Gender") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +

  scale_color_manual(values = c('cyan', 'magenta'))


plot.batch.sex




```


## Plot Age
```{r  fig.height=4.2, fig.width=5}
plot.batch.age <- 
      con_scz$plotGraph(groups = agefactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600,
        #raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Age") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2)) +

  scale_color_manual(values = color.age)


plot.batch.age




```




## Plot PMI

```{r  fig.height=4.2, fig.width=5}
plot.batch.pmi <- 
      con_scz$plotGraph(groups = pmifactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, 
        raster = T, raster.dpi = 600,
       # raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "PMI") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2)) +

  scale_color_manual(values = color.PMI)
 

plot.batch.pmi
```


## Plot Source

```{r  fig.height=4.2, fig.width=5}
plot.batch.source <- 
      con_scz$plotGraph(groups = sourcefactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = cols25(), 
        raster = T, raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Brain bank") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.source
```

## Plot cause of death

```{r fig.height=4.2, fig.width=6}
plot.batch.death <- 
      con_scz$plotGraph(groups = deathfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600,
       # raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Cause of death") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.death)
 

plot.batch.death

```



## Plot FACS date
```{r fig.height=4.2, fig.width=5}
plot.batch.facs <- 
      con_scz$plotGraph(groups = facsfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = cols25(), 
        raster = T, raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "FACS date") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.facs
```

## Plot % NeuN positive nuclei

```{r fig.height=4.2, fig.width=5}
plot.batch.neun <- 
      con_scz$plotGraph(groups = neunfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600,
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "% NeuN+ nuclei during FACS") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2)) +
  scale_color_manual(values = color.neun)
 

plot.batch.neun
```

## Plot GEM kit lot
```{r fig.height=4.2, fig.width=5}
plot.batch.gem <- 
      con_scz$plotGraph(groups = gemfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = rpal, 
        raster = T, raster.dpi = 600,
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "GEM kit lot") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.gem
```

## Plot preamp date
```{r fig.height=4.2, fig.width=5}
plot.batch.preamp <- 
      con_scz$plotGraph(groups = preampfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = rpal, 
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "cDNA pre-amplification date") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.preamp
```
## Plot lib prep date
```{r fig.height=4.2, fig.width=5}
plot.batch.libprep <- 
      con_scz$plotGraph(groups = libdatefactor[
        names(con_scz$getDatasetPerCell()) %in% names(annotations_scz2$high)],
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = rpal, 
        raster = T,  raster.dpi = 600,
       # raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Library prep date") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.libprep
```

## Plot lib prep kit lot
```{r fig.height=4.2, fig.width=5}
plot.batch.liblot <- 
      con_scz$plotGraph(groups = libfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = rpal, 
        raster = T,  raster.dpi = 600,
       # raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Library prep kit lot") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.liblot
```

## Plot volume of nuclei susp loaded

```{r fig.height=4.2, fig.width=5}
plot.batch.vol <- 
      con_scz$plotGraph(groups = volfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Nuclei suspension/10x rxn, ul") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2)) +
  scale_color_manual(values = color.vol)
 

plot.batch.vol
```

## Plot cDNA concentration

```{r fig.height=4.2, fig.width=5}
plot.batch.cdna <- 
      con_scz$plotGraph(groups = cdnafactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "cDNA concentration, ng/ul") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2)) +
  scale_color_manual(values = color.cdna)
 

plot.batch.cdna
```



## Plot amount DNA per lib prep

```{r fig.height=4.2, fig.width=4.2}
plot.batch.amountlib <- 
      con_scz$plotGraph(groups = amountlibfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = rpal,
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Amount cDNA/library prep, ng") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1))
 

plot.batch.amountlib
```




## Plot index primer

```{r fig.height=4.2, fig.width=5}
plot.batch.index <- 
      con_scz$plotGraph(groups = indexfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F, palette = rpal,
        raster = T,  raster.dpi = 600, 
        #raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Index primer") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=2))
 

plot.batch.index
```

## Plot hypoxia postmortem

```{r fig.height=4.2, fig.width=5}
plot.batch.hypoxia <- 
      con_scz$plotGraph(groups = hypoxiafactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Signs of postmortem brain hypoxia") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.hypoxia
```


## Plot tangles

```{r fig.height=4.2, fig.width=5}
plot.batch.tangles <- 
      con_scz$plotGraph(groups = tanglesfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Protein tangles") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.tangles)
 

plot.batch.tangles
```
## Plot amyloid

```{r fig.height=4.2, fig.width=5}
plot.batch.amyloid <- 
      con_scz$plotGraph(groups = amyloidfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600,
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Amyloid") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.amyloid)
 

plot.batch.amyloid
```

## Plot atherosclerosis

```{r fig.height=4.2, fig.width=5}
plot.batch.athero <- 
      con_scz$plotGraph(groups = atherofactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600,
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Atherosclerosis") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.athero)
 

plot.batch.athero
```


## Plot Lewy bodies/Parkinsonism

```{r fig.height=4.2, fig.width=5}
plot.batch.lewy <- 
      con_scz$plotGraph(groups = lewyfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Lewy bodies") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.lewy)
 

plot.batch.lewy
```



## Plot Lithium treatment

```{r fig.height=4.2, fig.width=5}
plot.batch.lithium <- 
      con_scz$plotGraph(groups = lithiumfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T,  raster.dpi = 600,
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Lithium treatment") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.lithium
```

## Plot Neuroleptic duration years

```{r fig.height=4.2, fig.width=5}
plot.batch.neurolept <- 
      con_scz$plotGraph(groups = neuroleptfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T,  raster.dpi = 600,
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Neuroleptics, years") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.neurolept)
 

plot.batch.neurolept
```


## typical nuroleptics

```{r fig.height=4.2, fig.width=5}
plot.batch.typical <- 
      con_scz$plotGraph(groups = typicalfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Typical neuroleptics") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.typical
```

## Atypical nuroleptics

```{r fig.height=4.2, fig.width=5}
plot.batch.atypical <- 
      con_scz$plotGraph(groups = atypicalfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
        #raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Atypical neuroleptics") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.atypical
```


## Benzodiazepines

```{r fig.height=4.2, fig.width=5}
plot.batch.benzo <- 
      con_scz$plotGraph(groups = benzofactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Benzodiazepines") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.benzo
```


## Barbiturates

```{r fig.height=4.2, fig.width=5}
plot.batch.barbi <- 
      con_scz$plotGraph(groups = barbifactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Barbiturates") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.barbi
```

## Tricyclic

```{r fig.height=4.2, fig.width=5}
plot.batch.tricyc <- 
      con_scz$plotGraph(groups = tricycfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Tricyclic antidepressants") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.tricyc
```


## SIRT/NE transporter inhibitors antidepressants

```{r fig.height=4.2, fig.width=5}
plot.batch.sirt <- 
      con_scz$plotGraph(groups = sirtfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "5-HT/NE transporters antidepressants") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.sirt
```

## Chemo tamoxifen

```{r fig.height=4.2, fig.width=5}
plot.batch.tamoxi <- 
      con_scz$plotGraph(groups = tamoxifactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Tamoxifen chemotherapy") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.tamoxi
```


## Chemo antiproliferative

```{r fig.height=4.2, fig.width=5}
plot.batch.prolif <- 
      con_scz$plotGraph(groups = proliffactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
       # raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Anti-proliferative chemotherapy") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.prolif
```

## Radiotherapy

```{r fig.height=4.2, fig.width=5}
plot.batch.radio <- 
      con_scz$plotGraph(groups = radiofactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
       # raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Radiotherapy") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.radio
```


## Cancer

```{r fig.height=4.2, fig.width=5}
plot.batch.cancer <- 
      con_scz$plotGraph(groups = cancerfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
   #     raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Cancer") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.cancer
```



## Onset age

```{r fig.height=4.2, fig.width=5}
plot.batch.onset <- 
      con_scz$plotGraph(groups = onsetfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "First psychiatric signs, age") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.onset)
 

plot.batch.onset
```


## Hypertension

```{r fig.height=4.2, fig.width=5}
plot.batch.hyper <- 
      con_scz$plotGraph(groups = hyperfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Hypertension") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.hyper
```

## Depression

```{r fig.height=4.2, fig.width=5}
plot.batch.depression <- 
      con_scz$plotGraph(groups = deprfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Depression") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.depression
```


## Malnutrition

```{r fig.height=4.2, fig.width=5}
plot.batch.malnut <- 
      con_scz$plotGraph(groups = malnutrfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Malnutrition/weight loss") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.malnut
```



## Breathing problems

```{r fig.height=4.2, fig.width=5}
plot.batch.breath <- 
      con_scz$plotGraph(groups = breathfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Breathing problems") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.breath
```


## Opioid

```{r fig.height=4.2, fig.width=5}
plot.batch.opioid <- 
      con_scz$plotGraph(groups = opioidfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
   #     raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Opioid modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.opioid
```


## Dopaminergic modulators

```{r fig.height=4.2, fig.width=5}
plot.batch.dopamine <- 
      con_scz$plotGraph(groups = dopaminefactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Dopaminergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.dopamine
```


## Serotonin

```{r fig.height=4.2, fig.width=5}
plot.batch.serotonine <- 
      con_scz$plotGraph(groups = serotoninefactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Serotoninergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.serotonine
```


## Glu-ergic modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.glu <- 
      con_scz$plotGraph(groups = glufactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Glutamatergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.glu
```


## GABA-ergic modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.gaba <- 
      con_scz$plotGraph(groups = gabafactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "GABAergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.gaba
```

## Histamine modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.hist <- 
      con_scz$plotGraph(groups = histfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Histaminergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.hist
```


## Muscarine modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.musc <- 
      con_scz$plotGraph(groups = muscarfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Muscarine receptors modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.musc
```


## Nicotine receptors modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.nico <- 
      con_scz$plotGraph(groups = nicofactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Nicotine receptors modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.nico
```

## Adren-ergic modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.adren <- 
      con_scz$plotGraph(groups = adrenfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Adrenergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.adren
```

## Adenosinergic modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.adeno <- 
      con_scz$plotGraph(groups = adenfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Purinergic modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.adeno
```

## Glycine recept modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.gly <- 
      con_scz$plotGraph(groups = glycinefactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Glycine receptors modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.gly
```

## Na modulators

```{r fig.height=4.2, fig.width=5}
plot.batch.na <- 
      con_scz$plotGraph(groups = nafactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
    #    raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Sodium channels modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.na
```


## K channel modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.k <- 
      con_scz$plotGraph(groups = kfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
     #   raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Potassium channels modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.k
```

## Ca channel modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.ca <- 
      con_scz$plotGraph(groups = cafactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
   #     raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Calcium channels modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.ca
```

## Proton pump modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.proton <- 
      con_scz$plotGraph(groups = protonfactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Proton pump modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.proton
```


## MAO modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.mao <- 
      con_scz$plotGraph(groups = maofactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "MAO modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.mao
```


## Mitho e transport modulation

```{r fig.height=4.2, fig.width=5}
plot.batch.mito <- 
      con_scz$plotGraph(groups = mitofactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
      #  raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Mitochondrial e transport modulators") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.mito
```

## Smoking

```{r fig.height=4.2, fig.width=5}
plot.batch.smoke <- 
      con_scz$plotGraph(groups = smokefactor,
                    alpha = 0.03, size = 0.1,
                    font.size = 4, show.legend = F, plot.na = F, mark.groups = F,
        raster = T, raster.dpi = 600, 
       # raster.height = 12, raster.width = 12
     
                    ) +
 
  theme_void() +
  labs(subtitle = "Smoking") +
  theme(text = element_text(family = "Liberation Sans"), legend.position = "right", 
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.title = element_blank(),
        legend.text = element_text(size = 12)) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2), ncol=1)) +
  scale_color_manual(values = color.yn)
 

plot.batch.smoke
```





## tSNE covariates 1

```{r fig.width=17.72, fig.height=21.65}
fig.s5_1 <-
  (plot.anno.high + theme(plot.margin = unit(c(0, 0.6, 0, 1.7), "cm"))) + 
  (plot.batch.samples + theme(plot.margin = unit(c(0, 0.6, 0, 0), "cm"))) + 
  (plot.batch.ph + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) +
  (plot.batch.neun + theme(plot.margin = unit(c(1.4, 0.6, 0, 1.7), "cm"))) +
  (plot.batch.disease + theme(plot.margin = unit(c(1.4, 0.6, 0, 0), "cm"))) +
  (plot.batch.sex + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.age + theme(plot.margin = unit(c(1.4, 0.6, 0, 1.7), "cm"))) +
  (plot.batch.pmi + theme(plot.margin = unit(c(1.4, 0.6, 0, 0), "cm"))) +
  (plot.batch.source + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.death + theme(plot.margin = unit(c(1.4, 0.6, 0, 1.7), "cm"))) + 
  (plot.batch.facs + theme(plot.margin = unit(c(1.4, 0.6, 0, 0), "cm"))) +
  (plot.batch.gem + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.preamp + theme(plot.margin = unit(c(1.4, 0.6, 0, 1.7), "cm"))) +
  (plot.batch.libprep + theme(plot.margin = unit(c(1.4, 0.6, 0, 0), "cm"))) +
  (plot.batch.liblot + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  
  plot_layout(ncol = 3) & theme(plot.subtitle = element_text(size = 20), legend.text = element_text(size = 17))
  

fig.s5_1
```

```{r}
ggsave("./plots/fig.s5_1.pdf", fig.s5_1, device = cairo_pdf, width = 17.72, height = 21.65, units = "in", dpi = 600)
```


## tSNE covariates 2

```{r fig.width=17.72, fig.height=21.65}
fig.s5_2 <-
  (plot.anno.high + theme(plot.margin = unit(c(0, 1.8, 0, 1.4), "cm"))) + 
  (plot.batch.vol + theme(plot.margin = unit(c(0, 1.8, 0, 0), "cm"))) +
  (plot.batch.cdna + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) +
  (plot.batch.amountlib + theme(plot.margin = unit(c(0.5, 1.8, 0, 1.4), "cm"))) + 
  (plot.batch.index + theme(plot.margin = unit(c(0.5, 1.8, 0, 0), "cm"))) +
  (plot.batch.hypoxia + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.tangles + theme(plot.margin = unit(c(0.5, 1.8, 0, 1.4), "cm"))) +
  (plot.batch.amyloid + theme(plot.margin = unit(c(0.5, 1.8, 0, 0), "cm"))) +
  (plot.batch.athero + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.lewy + theme(plot.margin = unit(c(0.5, 1.8, 0, 1.4), "cm"))) +
  (plot.batch.lithium + theme(plot.margin = unit(c(0.5, 1.8, 0, 0), "cm"))) +
  (plot.batch.neurolept + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
   (plot.batch.typical + theme(plot.margin = unit(c(0.5, 1.8, 0, 1.4), "cm"))) +
   (plot.batch.atypical + theme(plot.margin = unit(c(0.5, 1.8, 0, 0), "cm"))) +
   (plot.batch.benzo + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  
  plot_layout(ncol = 3) & theme(plot.subtitle = element_text(size = 20), legend.text = element_text(size = 17))
  

fig.s5_2
```

```{r}
ggsave("./plots/fig.s5_2.pdf", fig.s5_2, device = cairo_pdf, width = 17.72, height = 21.65, units = "in", dpi = 600)
```




## tSNE covariates 3

```{r fig.width=17.72, fig.height=21.65}
fig.s5_3 <-
  (plot.anno.high + theme(plot.margin = unit(c(0, 3, 0, 2.5), "cm"))) + 
  (plot.batch.barbi + theme(plot.margin = unit(c(0, 3, 0, 0), "cm"))) +
  (plot.batch.tricyc + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) +
  (plot.batch.sirt + theme(plot.margin = unit(c(0.5, 3, 0, 2.5), "cm"))) + 
  (plot.batch.tamoxi + theme(plot.margin = unit(c(0.5, 3, 0, 0), "cm"))) +
  (plot.batch.prolif + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.radio + theme(plot.margin = unit(c(0.5, 3, 0, 2.5), "cm"))) +
  (plot.batch.cancer + theme(plot.margin = unit(c(0.5, 3, 0, 0), "cm"))) +
  (plot.batch.hyper + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.depression + theme(plot.margin = unit(c(0.5, 3, 0, 2.5), "cm"))) +
  (plot.batch.malnut + theme(plot.margin = unit(c(0.5, 3, 0, 0), "cm"))) +
  (plot.batch.breath + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
   (plot.batch.opioid + theme(plot.margin = unit(c(0.5, 3, 0, 2.5), "cm"))) +
   (plot.batch.dopamine + theme(plot.margin = unit(c(0.5, 3, 0, 0), "cm"))) +
   (plot.batch.serotonine + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  
  
  plot_layout(ncol = 3) & theme(plot.subtitle = element_text(size = 20), legend.text = element_text(size = 17))
  

fig.s5_3
```

```{r}
ggsave("./plots/fig.s5_3.pdf", fig.s5_3, device = cairo_pdf, width = 17.72, height = 21.65, units = "in", dpi = 600)
```


## tSNE covariates 4

```{r fig.width=17.72, fig.height=21.65}
fig.s5_4 <-
  (plot.anno.high + theme(plot.margin = unit(c(0, 3.5, 0, 2.5), "cm"))) + 
  (plot.batch.glu + theme(plot.margin = unit(c(0, 3.5, 0, 0), "cm"))) +
  (plot.batch.gaba + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) +
  (plot.batch.hist + theme(plot.margin = unit(c(0.5, 3.5, 0, 2.5), "cm"))) + 
  (plot.batch.musc + theme(plot.margin = unit(c(0.5, 3.5, 0, 0), "cm"))) +
  (plot.batch.nico + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.adren + theme(plot.margin = unit(c(0.5, 3.5, 0, 2.5), "cm"))) +
  (plot.batch.adeno + theme(plot.margin = unit(c(0.5, 3.5, 0, 0), "cm"))) +
  (plot.batch.gly + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.na + theme(plot.margin = unit(c(0.5, 3.5, 0, 2.5), "cm"))) +
  (plot.batch.k + theme(plot.margin = unit(c(0.5, 3.5, 0, 0), "cm"))) +
  (plot.batch.ca + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.proton + theme(plot.margin = unit(c(0.5, 3.5, 0, 2.5), "cm"))) +
  (plot.batch.mao + theme(plot.margin = unit(c(0.5, 3.5, 0, 0), "cm"))) +
  (plot.batch.mito + theme(plot.margin = unit(c(0.5, 0, 0, 0), "cm"))) +
  (plot.batch.smoke + theme(plot.margin = unit(c(0.5, 3.5, 0, 2.5), "cm"))) +
  (plot.batch.onset + theme(plot.margin = unit(c(0.5, 3.5, 0, 0), "cm"))) +
  
  
  
  plot_layout(ncol = 3) & theme(plot.subtitle = element_text(size = 20), legend.text = element_text(size = 17))
  

fig.s5_4
```


```{r}
ggsave("./plots/fig.s5_4.pdf", fig.s5_4, device = cairo_pdf, width = 17.72, height = 21.65, units = "in", dpi = 600)
```





# tSNE covariates trimmed list
## 1

```{r fig.width=17.72, fig.height=21.65}
fig.s4_1 <-
  (plot.anno.high + theme(plot.margin = unit(c(0, 1.5, 0, 2), "cm"))) + 
  (plot.batch.samples + theme(plot.margin = unit(c(0, 1.5, 0, 0), "cm"))) + 
  (plot.batch.ph + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) +
  (plot.batch.neun + theme(plot.margin = unit(c(1.4, 1.5, 0, 2), "cm"))) +
  (plot.batch.disease + theme(plot.margin = unit(c(1.4, 1.5, 0, 0), "cm"))) +
  (plot.batch.sex + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.age + theme(plot.margin = unit(c(1.4, 1.5, 0, 2), "cm"))) +
  (plot.batch.pmi + theme(plot.margin = unit(c(1.4, 1.5, 0, 0), "cm"))) +
  (plot.batch.source + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.death + theme(plot.margin = unit(c(1.4, 1.5, 0, 2), "cm"))) + 
  (plot.batch.facs + theme(plot.margin = unit(c(1.4, 1.5, 0, 0), "cm"))) +
  (plot.batch.preamp + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.gem + theme(plot.margin = unit(c(1.4, 1.5, 0, 2), "cm"))) +
  (plot.batch.libprep + theme(plot.margin = unit(c(1.4, 1.5, 0, 0), "cm"))) +
  (plot.batch.liblot + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  (plot.batch.amyloid + theme(plot.margin = unit(c(1.4, 1.5, 0, 2), "cm"))) +
  (plot.batch.athero + theme(plot.margin = unit(c(1.4, 1.5, 0, 0), "cm"))) +
  (plot.batch.tangles + theme(plot.margin = unit(c(1.4, 0, 0, 0), "cm"))) +
  
  plot_layout(ncol = 3) & theme(plot.subtitle = element_text(size = 20), legend.text = element_text(size = 17))
  

fig.s4_1






```

```{r}
ggsave("./plots/fig.s4_1.pdf", fig.s4_1, device = cairo_pdf, width = 17.72, height = 21.65, units = "in", dpi = 600)
```



## 2

```{r fig.width=8.86, fig.height=21.65}
fig.s4_2 <-
  (plot.anno.high + theme(plot.margin = unit(c(0, 0.6, 0, 1.7), "cm"))) + 
  (plot.batch.typical + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) + 
  (plot.batch.atypical + theme(plot.margin = unit(c(1.9, 0.6, 0, 1.7), "cm"))) +
  (plot.batch.neurolept + theme(plot.margin = unit(c(1.9, 0.6, 0, 0), "cm"))) +
  (plot.batch.benzo + theme(plot.margin = unit(c(1.9, 0.6, 0, 1.7), "cm"))) +
  
  (plot.batch.glu + theme(plot.margin = unit(c(1.9, 0.6, 0, 0), "cm"))) +
  (plot.batch.gaba + theme(plot.margin = unit(c(1.9, 0.6, 0, 1.7), "cm"))) +
  
  (plot.batch.amyloid + theme(plot.margin = unit(c(1.9, 0.6, 0, 0), "cm"))) + 
  (plot.batch.athero + theme(plot.margin = unit(c(1.9, 0.6, 0, 1.7), "cm"))) +
  (plot.batch.tangles + theme(plot.margin = unit(c(1.9, 0.6, 0, 0), "cm"))) +

  
  plot_layout(ncol = 2) & theme(plot.subtitle = element_text(size = 20), legend.text = element_text(size = 17))
  

fig.s4_2



```



```{r}
ggsave("./plots/fig.s4_2.pdf", fig.s4_2, device = cairo_pdf, width = 8.86, height = 21.65, units = "in", dpi = 300)
```




# Marker expression for IHC/RNAScope image




## Plot Annotations High
```{r fig.height=6, fig.width=6}




plot.anno.high2 <- 
      con_scz$plotGraph(groups = annotations_scz2$high,
                    alpha = 0.2, size = 0.5,
                    font.size = 3, show.legend = F, plot.na = F, mark.groups = T, 
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Subtypes, high resolution") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none") +
  scale_color_manual(values = high.pal)


plot.anno.high2

```


```{r}
ggsave("./plots/plot.anno.high2.pdf", plot.anno.high2, device = cairo_pdf, width = 6, height = 6, units = "in", dpi = 600)
```



## Plot Annotations Med
```{r fig.height=6, fig.width=6}




plot.anno.med <- 
      con_scz$plotGraph(groups = annotations_scz2$med,
                    alpha = 0.2, size = 0.5,
                    font.size = 3, show.legend = F, plot.na = F, mark.groups = T, 
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Subtypes, medium resolution") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none") +
  scale_color_manual(values = med.pal)


plot.anno.med

```

## Plot VIP
```{r fig.height=6, fig.width=6}




plot.vip <- 
      con_scz$plotGraph(gene = "VIP",
                    alpha = 0.4, size = 0.5,
                    font.size = 4, show.legend = F, plot.na = F,
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "VIP") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")


plot.vip

```

## Plot CALB2
```{r fig.height=6, fig.width=6}




plot.calb2 <- 
      con_scz$plotGraph(gene = "CALB2",
                    alpha = 0.4, size = 0.5,
                    font.size = 4, show.legend = F, plot.na = F,
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "CALB2(Calretinin)") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")


plot.calb2

```


## Plot PV
```{r fig.height=6, fig.width=6}




plot.PV <- 
      con_scz$plotGraph(gene = "PVALB",
                    alpha = 0.4, size = 0.5,
                    font.size = 4, show.legend = F, plot.na = F,
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "PVALB(PV)") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")


plot.PV

```



## Plot CRH
```{r fig.height=6, fig.width=6}




plot.CRH <- 
      con_scz$plotGraph(gene = "CRH",
                    alpha = 0.4, size = 0.5,
                    font.size = 4, show.legend = F, plot.na = F,
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "CRH") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")


plot.CRH

```



## Plot CALB1
```{r fig.height=6, fig.width=6}




plot.CALB1 <- 
      con_scz$plotGraph(gene = "CALB1",
                    alpha = 0.4, size = 0.5,
                    font.size = 4, show.legend = F, plot.na = F,
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "CALB1(Calbindin)") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")


plot.CALB1

```


## Plot NPY
```{r fig.height=6, fig.width=6}




plot.NPY <- 
      con_scz$plotGraph(gene = "NPY",
                    alpha = 0.4, size = 0.5,
                    font.size = 4, show.legend = F, plot.na = F,
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "NPY") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")


plot.NPY

```

## Markers_for_IHC_RNAScope

```{r fig.width=12.76, fig.height=14.2}
fig.s_markers_IHC <-
  (plot.anno.med + theme(plot.margin = unit(c(0, 0.7, 0, 0), "cm"))) + 
  (plot.anno.high2 + theme(plot.margin = unit(c(0, 0.7, 0, 0), "cm"))) +
  (plot.vip + theme(plot.margin = unit(c(0.7, 0, 0, 0), "cm"))) +
  (plot.calb2 + theme(plot.margin = unit(c(0.7, 0.7, 0, 0), "cm"))) + 
  (plot.CRH + theme(plot.margin = unit(c(0.7, 0.7, 0, 0), "cm"))) +
  (plot.CALB1 + theme(plot.margin = unit(c(0.7, 0, 0, 0), "cm"))) +
  (plot.PV + theme(plot.margin = unit(c(0.7, 0.7, 0, 0), "cm"))) +
  (plot.NPY + theme(plot.margin = unit(c(0.7, 0.7, 0, 0), "cm"))) +
  
  
  
  
  plot_layout(ncol = 3)
  

fig.s_markers_IHC
```

```{r}
ggsave("./plots/fig.s_markers_IHC.pdf", fig.s_markers_IHC, device = cairo_pdf, width = 12.76, height = 14.2, units = "in", dpi = 600)
```






# Parts Fig 1

## Plot Annotations High
```{r fig.height=5, fig.width=3.9}




plot.anno.high3 <- 
      con_scz$plotGraph(groups = annotations_scz2$high,
                    alpha = 0.2, size = 0.5,
                    font.size = 2.5, show.legend = F, plot.na = F, mark.groups = T, 
        raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
  labs(subtitle = "Subtypes, high resolution") +
  theme(text = element_text(family = "Liberation Sans"),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none") +
  scale_color_manual(values = high.pal)


plot.anno.high3

```


```{r}
ggsave("./plots/plot.anno.high3.pdf", plot.anno.high3, device = cairo_pdf, width = 3.9, height = 5, units = "in", dpi = 600)
```





## Umap with marker expression



```{r fig.width=3.9, fig.height=3.9}
plot.list.markers <- lapply(X = c("GAD1", "SATB2", "CUX2", "RORB", "FEZF2", "THEMIS", "VIP",
                                  "ID2", "SST", "PVALB", "SLC1A3", "MBP"),
                               function(x) {
  con_scz$plotGraph(gene = x, title = x, alpha= 1, size = 0.3, plot.na = F, raster = T, 
        raster.height = 12, raster.width = 12
                    ) +
 
  theme_void() +
 
  theme(text = element_text(family = "Arial"),
        plot.title = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position="none")
                    
                    
                    
}
)


plot.markers <- grid.arrange(grobs = plot.list.markers, ncol = 4)



```

```{r}
ggsave("./plots/plot.markers.pdf", plot.markers, device = cairo_pdf, width = 3.9, height = 3.9, units = "in", dpi = 600)


```



## Layer distribution for MTG

### import data
```{r}

annot_tranfer_mtg_highres <-

  read_rds("/home/mbatiuk/projects/human_schizo/MB6-MB23/2ndary/ctx_layers_SCZ_ALLEN/rds_objects/mtg_high_transf_annotations.rds")



```


### Plot layer MTG, remove "Other" from plots - lower quality nucs
  
```{r fig.width=4.25, fig.height=4.7}
paper.layers.mtg.high <-
 ggplot(

  data.frame(
       Subtypes = annot_tranfer_mtg_highres$labels[
            annotation_allen$sample_name[annotation_allen$sample_name %in% names(annot_tranfer_mtg_highres$labels)]
            ] %>%
       
        `[`(., .!= "Other") %>%
         factor(levels = 
          c("L2_CUX2_LAMP5_MARCH1", "L2_CUX2_LAMP5_PDGFD", "L2_3_CUX2_FREM3_UNC5D", "L2_3_CUX2_FREM3_SV2C",
            "L3_CUX2_PRSS12", "L4_RORB_SCHLAP1_MET", "L4_RORB_SCHLAP1_MME", "L4_RORB_SCHLAP1_ARHGAP15", "L4_5_FEZF2_LRRK1",
            "L5_FEZF2_ADRA1A", "L5_6_FEZF2_TLE4_ABO", "L5_6_FEZF2_TLE4_SCUBE1", "L5_6_FEZF2_TLE4_HTR2C", "L5_6_THEMIS_SEMA3A",
            "L5_6_THEMIS_NTNG2", "ID2_LAMP5_NMBR", "ID2_LAMP5_CRH", "ID2_LAMP5_NOS1", "ID2_PAX6", "ID2_NCKAP5", 
            "VIP_ABI3BP", "VIP_TYR", "VIP_RELN", "VIP_SEMA3C", "VIP_SSTR1", "VIP_CRH", "PVALB_MEPE", "PVALB_SST", 
            "PVALB_CRH", "SST_TH", "SST_TAC3", "SST_CALB1", "SST_NPY", "SST_NOS1", "SST_STK32A", "Glia")
          ),
        
        Layers = setNames(annotation_allen$cortical_layer_label, annotation_allen$sample_name)[
                            
          
          annot_tranfer_mtg_highres$labels[
            annotation_allen$sample_name[annotation_allen$sample_name %in% names(annot_tranfer_mtg_highres$labels)]
            ] %>%
       
        `[`(., .!= "Other") %>% names
        
        
        ] %>%
         
         
         
         
         factor(levels = c("L6", "L5", "L4", "L3", "L2", "L1"))
        ), aes(x = Subtypes, y = Layers,  color = Subtypes)
  ) +
  geom_jitter(width = 0.35, height = 0.4, alpha = 0.5, size = 0.1, show.legend = F) +
  geom_hline(yintercept = c(1.5, 2.5, 3.5, 4.5, 5.5), linetype = 2, size = 1, colour = "grey") +
  scale_color_manual(values = high.pal[levels(annotations_scz2$high)]) +
  labs(title = "Position, MTG estimate, high resolution") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 7), 
        plot.title = element_text(hjust = 0.5, size = 15, face = "bold"),
        axis.text.y = element_text(size = 12), axis.title = element_text(size = 12),
        text = element_text(family = "Liberation Sans"), axis.title.x = element_blank(),
        axis.line = element_line(size = 0.8),
        axis.ticks = element_blank())


paper.layers.mtg.high



```


```{r}
ggsave("./plots/paper.layers.mtg.high.pdf", paper.layers.mtg.high, device = cairo_pdf, width = 4.25, height = 4.7, 
       units = "in", dpi = 600)


```




### rasterized version

```{r fig.width=4.25, fig.height=4.7}
paper.layers.mtg.high.rast <-
 ggplot(

  data.frame(
       Subtypes = annot_tranfer_mtg_highres$labels[
            annotation_allen$sample_name[annotation_allen$sample_name %in% names(annot_tranfer_mtg_highres$labels)]
            ] %>%
       
        `[`(., .!= "Other") %>%
         factor(levels = 
          c("L2_CUX2_LAMP5_MARCH1", "L2_CUX2_LAMP5_PDGFD", "L2_3_CUX2_FREM3_UNC5D", "L2_3_CUX2_FREM3_SV2C",
            "L3_CUX2_PRSS12", "L4_RORB_SCHLAP1_MET", "L4_RORB_SCHLAP1_MME", "L4_RORB_SCHLAP1_ARHGAP15", "L4_5_FEZF2_LRRK1",
            "L5_FEZF2_ADRA1A", "L5_6_FEZF2_TLE4_ABO", "L5_6_FEZF2_TLE4_SCUBE1", "L5_6_FEZF2_TLE4_HTR2C", "L5_6_THEMIS_SEMA3A",
            "L5_6_THEMIS_NTNG2", "ID2_LAMP5_NMBR", "ID2_LAMP5_CRH", "ID2_LAMP5_NOS1", "ID2_PAX6", "ID2_NCKAP5", 
            "VIP_ABI3BP", "VIP_TYR", "VIP_RELN", "VIP_SEMA3C", "VIP_SSTR1", "VIP_CRH", "PVALB_MEPE", "PVALB_SST", 
            "PVALB_CRH", "SST_TH", "SST_TAC3", "SST_CALB1", "SST_NPY", "SST_NOS1", "SST_STK32A", "Glia")
          ),
        
        Layers = setNames(annotation_allen$cortical_layer_label, annotation_allen$sample_name)[
                            
          
          annot_tranfer_mtg_highres$labels[
            annotation_allen$sample_name[annotation_allen$sample_name %in% names(annot_tranfer_mtg_highres$labels)]
            ] %>%
       
        `[`(., .!= "Other") %>% names
        
        
        ] %>%
         
         
         
         
         factor(levels = c("L6", "L5", "L4", "L3", "L2", "L1"))
        ), aes(x = Subtypes, y = Layers,  color = Subtypes)
  ) +
  geom_jitter_rast(width = 0.35, height = 0.4, alpha = 0.5, size = 0.8, show.legend = F, 
                   raster.width = 9.19, raster.height = 6.72, 
                   raster.dpi = 600
                   ) +
  geom_hline(yintercept = c(1.5, 2.5, 3.5, 4.5, 5.5), linetype = 2, size = 1, colour = "grey") +
  scale_color_manual(values = high.pal[levels(annotations_scz2$high)]) +
  labs(title = "Estimated position, high resolution") +
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 7), 
        plot.title = element_text(hjust = 0.5, size = 15, face = "bold"),
        axis.text.y = element_text(size = 12), axis.title = element_text(size = 12),
        text = element_text(family = "Liberation Sans"), axis.title.x = element_blank(),
        axis.line = element_line(size = 0.8),
        axis.ticks = element_blank())


paper.layers.mtg.high.rast



```







```{r}
ggsave("./plots/paper.layers.mtg.high.rast.pdf", paper.layers.mtg.high.rast, device = cairo_pdf, width = 4.25, height = 4.7, 
       units = "in", dpi = 600)


```






# PSEUDOBULK Plots

## Get counts per sample


```{r}

#Using built-in functions of conos
#this gets raw cluster count matrices
#aggr1 <- con_scz$getClusterCountMatrices(common.genes = T, omit.na.cells = T, groups = annotations_scz2$high.clean)

#this gets raw matrices per sample and collapses them by cluster
#min.cell.count - dangerous - skips subtypes in some samples (rare subtypes) - but reduces noise
#use 10 cells per subtype/sample filter

aggr <- conos:::rawMatricesWithCommonGenes(con_scz) %>%
    lapply(conos:::collapseCellsByType, groups = annotations_scz2_no_MB8$high.clean, min.cell.count = 10) %>% lapply(t)






# Manual way. Start with import of data from h5 10x count matrices - MB8 excluded
cms <- lapply(setNames(names(diseasef_no_MB8), names(diseasef_no_MB8)), function(smplname){
             cm <- Seurat::Read10X_h5(
                paste0("/home/mbatiuk/projects/human_schizo/MB6-MB23/counts_NON_FILT_GENOME_ENSEMBLE_nova_run1_run2/",
                       smplname, "/outs/filtered_feature_bc_matrix.h5"))
             
             colnames(cm) <- paste0(smplname, "_", colnames(cm))
             
             cm
  
  
  
  
}
)




# filter nuclei - from annotations w/o Other group


cms <- lapply(cms, function(sample) {sample[, colnames(sample) %in% names(annotations_scz2$high.clean)]
}
)





#sum of cells in filtered cms
lapply(cms, function(x) dim(x)[2]) %>% unlist %>% sum


```


## Create Pseudobulk per sample per subtype (high resolution) NORMALIZED BY LIB SIZE: https://jef.works/blog/2020/04/06/quickly-creating-pseudobulks/

```{r}


cms.psbulk <- lapply(setNames(names(cms), names(cms)), function(smpl) {
  
  lapply(setNames(levels(annotations_scz2_no_MB8$high.clean), levels(annotations_scz2_no_MB8$high.clean)), function(subt) {
    
#get nuclei names for sample/subtype combination    
  cells.subt <- split(annotations_scz2_no_MB8$high.clean, 
          
          gsub("(^MB[[:digit:]][[:digit:]]?-?[[:digit:]]?)(.*)", "\\1",
                                           names(annotations_scz2_no_MB8$high.clean))
          
          )
  
  
  cells <- cells.subt[[smpl]][cells.subt[[smpl]] == subt] %>% names
  
  
  #filter cell/subtype combinations with < 10 cells + get pseudobulk
  if(length(cells) >= 10) {  
    
  pseudobulk <- rowSums(cms[[smpl]][, cells])
  
  #normalize
  pseudobulk * 1000000 /sum(pseudobulk)
  
 
  }  
       
  } 
  ) %>% 
    
    #remove NULL objects and merge to matrix
    plyr::compact() %>% sapply(function(x) x)
    
   
}
)


#save object
#write_rds(cms.psbulk, "./rds_objects/cms.psbulk.rds")





```




## make data frame

```{r}
psbulk.df <-

lapply(setNames(names(cms.psbulk), names(cms.psbulk)), function(smpl) {
  
  lapply(setNames(cms.psbulk[[smpl]] %>% colnames, cms.psbulk[[smpl]] %>% colnames), function(subt) {
    
    data.frame(sample = smpl, subtype = subt, condition = as.character(diseasef[smpl]), 
               t(cms.psbulk[[smpl]][, subt]))
    
    
  }
  ) %>% do.call(rbind, .)
  
}
) %>% do.call(rbind, .)

  
  


#fix factor orders

psbulk.df$subtype <- factor(psbulk.df$subtype, levels = hsub_order)


psbulk.df$condition <- factor(psbulk.df$condition, levels = c("control", "schizo"))
psbulk.df$condition %<>% gsub("schizo", "SZ", .) %>% factor


#save object

write_rds(psbulk.df, "./rds_objects/psbulk.df.rds")


```


## Log1p pseudobulk data frame

```{r}
psbulk.log1p.df <- psbulk.df

psbulk.log1p.df[, 4:dim(psbulk.log1p.df)[2]] %<>% log1p

#backup
#write_rds(psbulk.log1p.df, "./rds_objects/psbulk.log1p.df.rds")

```



# Pseudobulk gene of interest expressions

```{r}

write_xlsx(
psbulk.df[, c("sample", "subtype", "condition", "OXTR", "DRD2", "GABRA1",
              "GABRB2", "GABRD", "GRIK3", "GRIN2A", "GRIN2C", "CALB1",
              "CALB2", "PVALB", "SV2C"
          )], path = "./istvan_genes_01.2021.xlsx")

```




## DE DATA IMPORT - WALD TEST RESULTS FROM PETER

```{r}

de.high.wald <- read_rds("../peter_analysis/de.high.wald.rds")

de.high.wald[["ID2_LAMP5_CRH"]]$res["HES4", "pvalue"]
```






## Plot Gene expressions
```{r fig.width=8.5, fig.height=5.5}

lapply(kostya_genes, function(G) {

  temp.plot <-
 

  ggplot(data = psbulk.df, aes_(y = as.name(G), x = ~subtype, dodge = ~condition, fill = ~condition)) +
  
  
  
    
  
  geom_point(aes(color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.4, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.7
               ) +
    
    stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.7), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          axis.text.y=element_text(size = 15),
          axis.title.x = element_blank(),
          text = element_text(family = "Helvetica", size = 15),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Normalized UMIs", subtitle = G) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)])



ggsave(temp.plot, file = paste0("./Gene_plots_psbulk/", G,"_psbulk_plot.pdf"), width = 8.5, height = 5.5, 
         units = "in", dpi = 600)  
  
}
  
)



```


## Plot Gene expressions
```{r fig.width=8.5, fig.height=5.5}

lapply(c("CALB2", "VIP", "CRH", "OXTR", "DRD2"), function(G) {

  temp.plot <-
 

  ggplot(data = psbulk.df, aes_(y = as.name(G), x = ~subtype, dodge = ~condition, fill = ~condition)) +
  
  
  
    
  
  geom_point(aes(color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.4, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.7
               ) +
    
    stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.7), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          axis.text.y=element_text(size = 15),
          axis.title.x = element_blank(),
          text = element_text(family = "Helvetica", size = 15),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Normalized UMIs", subtitle = G) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)])



ggsave(temp.plot, file = paste0("./Gene_plots_psbulk/", G,"_psbulk_plot.pdf"), width = 8.5, height = 5.5, 
         units = "in", dpi = 600)  
  
}
  
)



```



## Plot gene expression log1p
```{r fig.width=8.5, fig.height=5.5}

lapply(kostya_genes, function(G) {

  temp.plot <-
 

  ggplot(data = psbulk.df, aes_(y = as.name(G), x = ~subtype, dodge = ~condition, fill = ~condition)) +
  
  
  
    
  
  geom_point(aes(color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.4, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.7
               ) +
    
    stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.7), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          axis.text.y=element_text(size = 15),
          axis.title.x = element_blank(),
          text = element_text(family = "Helvetica", size = 15),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "log(1 + normalized UMI)", subtitle = G) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_continuous(trans = "log1p")



ggsave(temp.plot, file = paste0("./Gene_plots_log_psbulk/", G,"_log_psbulk_plot.pdf"), width = 8.5, height = 5.5, 
         units = "in", dpi = 600)  
  
}
  
)



```







## Define inhibitory subtypes

```{r}
inhib.subs <- 
  psbulk.df[grepl("^ID2|^SST|^PVALB|^VIP", psbulk.df$subtype), ]$subtype %>% unique %>% sort %>% droplevels




```


## CHRFAM7A plot fig4
 
```{r fig.width=4.25, fig.height=3.2}

pvals <-
    lapply(setNames(as.vector(inhib.subs), inhib.subs), function(x) {
    de.high.wald[[x]]$res["CHRFAM7A", "pvalue"]
    }
    ) %>% unlist(recursive = F)


#correct p-vals for  multiple comparisons
pvals %<>% p.adjust(method = "BH")






CHRFAM7A.plot <-

ggplot(data = psbulk.df[psbulk.df$subtype %in% as.vector(inhib.subs), ], 
       aes(x = subtype, y = CHRFAM7A, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = CHRFAM7A, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.8
               ) +
 
  
  stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.8), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
   geom_signif(annotations = null.length.signif(pvals),
             xmin = null.length(c(0.75:19.75), pvals), 
             xmax = null.length(c(1.25:20.25), pvals), 
             y_position = null.length(rep(
               max(
     psbulk.df[psbulk.df$subtype %in% as.vector(inhib.subs), ][["CHRFAM7A"]]) * 1.05, 20), pvals),
             textsize = 5#, angle = 90, vjust = 0.7, hjust = -0.1
              ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_blank(),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top", panel.grid.major.x = element_blank(),
         
          
          
          
          plot.subtitle = element_text(size = 14, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -3.7, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = "Normalized UMIs", subtitle = "CHRFAM7A") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
   ylim(-0.2, max(
     psbulk.df[psbulk.df$subtype %in% as.vector(inhib.subs), ][["CHRFAM7A"]]) * 1.25)

CHRFAM7A.plot




```



```{r fig.width=4.25, fig.height=3.2}

pvals <-
    lapply(setNames(as.vector(inhib.subs), inhib.subs), function(x) {
    de.high.wald[[x]]$res["CHRFAM7A", "pvalue"]
    }
    ) %>% unlist(recursive = F)


#correct for multiple comparison


pvals %<>% p.adjust(method = "BH")


CHRFAM7A.log.plot <-

ggplot(data = psbulk.log1p.df[psbulk.log1p.df$subtype %in% as.vector(inhib.subs), ], 
       aes(x = subtype, y = CHRFAM7A, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = CHRFAM7A, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  
#stat_boxplot(geom ='errorbar') +
  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.8
               ) +
 
  
  stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.8), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
   geom_signif(annotations = null.length.signif(pvals),
             xmin = null.length(c(0.75:19.75), pvals), 
             xmax = null.length(c(1.25:20.25), pvals), 
             y_position = null.length(rep(
               max(
     psbulk.log1p.df[psbulk.log1p.df$subtype %in% as.vector(inhib.subs), ][["CHRFAM7A"]]) * 
       1.07, 20), pvals),
             textsize = 5#, angle = 90, vjust = 0.7, hjust = -0.1
              ) +
  
  
  
  
 
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_blank(),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
         
        
          
          
          plot.subtitle = element_text(size = 14, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -3.7, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = "log(1 + UMI count)", subtitle = "CHRFAM7A") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
   ylim(-0.2, max(
     psbulk.log1p.df[psbulk.log1p.df$subtype %in% as.vector(inhib.subs), ][["CHRFAM7A"]]) * 1.25)

CHRFAM7A.log.plot




```








```{r}
ggsave("~/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/fig4.CHRFAM7A.pdf", 
       CHRFAM7A.plot, width = 4.25, height = 3.2, dpi = 600, device = cairo_pdf)
```

## Pseudobulk plots


### null.length function - to make NULL if annotation vector has 0 lenght

```{r}

null.length.signif <- function(x) {
  
  if (signif(x)[
        x <= 0.05 & !is.na(x)] %>% length > 0) {
  
  
  signif(x)[
        x <= 0.05 & !is.na(x)]
  
  
} else {
          
  NULL
        }

  
}
  





null.length <- function(vect, x) {
  
  if (x[
        x <= 0.05 & !is.na(x)] %>% length > 0) {
  
  
  vect[
        x <= 0.05 & !is.na(x)]
  
  
} else {
          
  NULL
        }

  
}





```




Genes to plot S12: DRD2 GABARA1 GRIK3 GRIN2A GRIN2C OXTR

### log1p
```{r fig.width=4.25, fig.height=3.2}

lapply(c("DRD2", "GABRA1", "GRIK3", "GRIN2A", "GRIN2C", "OXTR"), function(G) {

  




ps <-
    lapply(setNames(as.vector(inhib.subs), inhib.subs), function(x) {
    de.high.wald[[x]]$res[G, "pvalue"]
    }
    ) %>% unlist(recursive = F)


#correct p-vals for  multiple comparisons
ps %<>% p.adjust(method = "BH")


temp.plot <-




ggplot(data = psbulk.log1p.df[psbulk.log1p.df$subtype %in% as.vector(inhib.subs), ], 
       aes_(y = as.name(G), x = ~subtype, dodge = ~condition, fill = ~condition)) +
  
  
  
    
  
  geom_point(aes(color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.8
               ) +
 
  
  stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.8), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
   geom_signif(annotations = null.length.signif(ps),
             xmin = null.length(c(0.75:19.75), ps), 
             xmax = null.length(c(1.25:20.25), ps), 
             y_position = null.length(rep(
               max(
     psbulk.log1p.df[psbulk.log1p.df$subtype %in% as.vector(inhib.subs), ][[G]]) * 1.07, 20), ps),
             textsize = 5#, angle = 90, vjust = 0.7, hjust = -0.1
              ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_blank(),
          text = element_text(family = "Helvetica", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
         
          
          
          
          plot.subtitle = element_text(size = 14, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -3.7, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = "log(1 + UMI count)", subtitle = G) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
   ylim(-0.2, max(
     psbulk.log1p.df[psbulk.log1p.df$subtype %in% as.vector(inhib.subs), ][[G]]) * 1.25)





ggsave(temp.plot, file = paste0("./Gene_plots_log_psbulk/", G,"_log_psbulk_plot.pdf"), width = 4.25, height = 3.2, 
         units = "in", dpi = 600)  
  
}
  
)




```



```{r}
#list p-values

ps <-

lapply(setNames(c("DRD2", "GABRA1", "GRIK3", "GRIN2A", "GRIN2C", "OXTR"),
                c("DRD2", "GABRA1", "GRIK3", "GRIN2A", "GRIN2C", "OXTR")), function(G) {

  




ps <-
    lapply(setNames(as.vector(inhib.subs), inhib.subs), function(x) {
    de.high.wald[[x]]$res[G, "pvalue"]
    }
    ) %>% unlist(recursive = F)


#correct p-vals for  multiple comparisons
ps %<>% p.adjust(method = "BH")
})
```



### linear
```{r fig.width=4.25, fig.height=3.2}

lapply(c("DRD2", "GABRA1", "GRIK3", "GRIN2A", "GRIN2C", "OXTR"), function(G) {

  




ps <-
    lapply(setNames(as.vector(inhib.subs), inhib.subs), function(x) {
    de.high.wald[[x]]$res[G, "pvalue"]
    }
    ) %>% unlist(recursive = F)

ps %<>% p.adjust(method = "BH")

temp.plot <-




ggplot(data = psbulk.df[psbulk.df$subtype %in% as.vector(inhib.subs), ], 
       aes_(y = as.name(G), x = ~subtype, dodge = ~condition, fill = ~condition)) +
  
  
  
    
  
  geom_point(aes(color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.4, lwd = 0.4, width = 0.8
               ) +
 
  
  stat_summary(fun = "median", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.8), 
                 mapping = aes(group = condition), fill = "grey",
                 shape = 23
           ) + 
  
   geom_signif(annotations = null.length.signif(ps),
             xmin = null.length(c(0.75:19.75), ps), 
             xmax = null.length(c(1.25:20.25), ps), 
             y_position = null.length(rep(
               max(
     psbulk.df[psbulk.df$subtype %in% as.vector(inhib.subs), ][[G]]) * 1.07, 20), ps),
             textsize = 5#, angle = 90, vjust = 0.7, hjust = -0.1
              ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_blank(),
          text = element_text(family = "Helvetica", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top", panel.grid.major.x = element_blank(),
          
         
          
          
          
          plot.subtitle = element_text(size = 14, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -3.7, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = "UMI counts", subtitle = G) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
   ylim(-0.2, max(
     psbulk.df[psbulk.df$subtype %in% as.vector(inhib.subs), ][[G]]) * 1.25)





ggsave(temp.plot, file = paste0("./Gene_plots_psbulk/", G,"_psbulk_plot.pdf"), width = 4.25, height = 3.2, 
         units = "in", dpi = 600)  
  
}
  
)




```






# Amount of nuclei in dataset and genes in dataset




## import data, prep data.frames

```{r}

#Import Scz datasets and give names to the elements
cms_scz_list <- lapply(setNames(unlist(samplegroups), unlist(samplegroups)),
                                
                                function(sn){
    Seurat::Read10X_h5(
      paste0(
        "~/projects/human_schizo/MB6-MB23/counts_NON_FILT_GENOME_ENSEMBLE_nova_run1_run2/", sn,
        "/outs/filtered_feature_bc_matrix.h5"
        )
    )
  }
  )
  



#rename nuclei names in matrix, prepend with sample name intead of "one"

cms_scz_list <- lapply(setNames(unlist(samplegroups), unlist(samplegroups)),
     function(sn) {
       paste0(sn, "_", colnames(cms_scz_list[[sn]])) %>%
         set_colnames(cms_scz_list[[sn]], .)
                         
      
      
     }
   )


#Filter out cells in scz samples and not present in annotation

cms_scz_list %<>% lapply(function(smpl) {
  smpl[, colnames(smpl) %in% names(annotations_scz2$high)]
    }
  )



#Count amount of expressed genes/cell >0


genes_per_nuc <- 

lapply(cms_scz_list, function(y) {apply(y, MARGIN = 2, FUN = function(x) {
  sum(x > 0)
  
  
})
}
  )



genes_per_nuc_vec <-

    setNames(unlist(genes_per_nuc),
    
    gsub("^MB[[:digit:]][[:digit:]]?-?[[:digit:]]?.", "", names(unlist(genes_per_nuc)))
    


)


#now for umis

umi_per_nuc <- 

lapply(cms_scz_list, function(y) {apply(y, MARGIN = 2, FUN = function(x) {
  sum(x)
  
  
})
}
  )


umi_per_nuc_vec <-

    setNames(unlist(umi_per_nuc),
    
    gsub("^MB[[:digit:]][[:digit:]]?-?[[:digit:]]?.", "", names(unlist(umi_per_nuc)))
    


)





#remove cms_scz_list to save ram

#rm(cms_scz_list)




#make data.frame


genes.p.nuc.df <- data.frame(genes_expressed = genes_per_nuc_vec,
                             sample = gsub("(^MB[[:digit:]][[:digit:]]?-?[[:digit:]]?)(.*)", "\\1",
                                           names(genes_per_nuc_vec)),
                                           
                                           condition = diseasef[
                                             gsub("(^MB[[:digit:]][[:digit:]]?-?[[:digit:]]?)(.*)", "\\1",
                                           names(genes_per_nuc_vec))
                                             
                                             
                                           ],
                             
                             subtype = annotations_scz2$high[names(genes_per_nuc_vec)],
                             
                             umis_detected = umi_per_nuc_vec[names(genes_per_nuc_vec)]
)




genes.p.nuc.df$sample %<>% factor(levels = names(diseasef))

genes.p.nuc.df$subtype %<>% factor(levels = hsub_order)


#get # samples per subtype


samp.p.subt.df <-
 
  genes.p.nuc.df %>% group_by(sample, subtype) %>% summarize(n_of_cells = n())

#filter subtypes with less than 5 or 10 cells

samp.p.subt.df <- samp.p.subt.df[samp.p.subt.df$n_of_cells >= 5, ]
 
#summarize

samp.p.subt.df %<>% ungroup %>% group_by(subtype) %>% summarize(freq = n()/length(diseasef))




#get # of subtypes per sample



subs.p.sample.df <-
 
  genes.p.nuc.df %>% group_by(sample, subtype) %>% summarize(n_of_cells = n())

#filter subtypes with less than 5 or 10 cells

subs.p.sample.df <- subs.p.sample.df[subs.p.sample.df$n_of_cells >= 5, ]
 
#summarize

subs.p.sample.df %<>% ungroup %>% group_by(sample) %>% summarize(freq = n()/length(annotations_scz2$high %>% unique))


subs.p.sample.df$condition <- diseasef[subs.p.sample.df$sample]




#nuclei per sample

nuc.p.sample.df <-

    genes.p.nuc.df %>% group_by(sample, subtype) %>% summarize(count = n()) %>% 
      group_by(subtype) %>% mutate(fraction = count/sum(count))




nuc.p.condition.df <-
  
    genes.p.nuc.df %>% group_by(condition, subtype) %>% summarize(count = n()) %>% 
      group_by(subtype) %>% mutate(fraction = count/sum(count))




```

### SUPPL DATA nuclei genes and umis per sample

```{r}

#genes/umis per sample
 
gene.umi.sample <- genes.p.nuc.df %>% group_by(sample) %>% 
  summarize_at(c("genes_expressed", "umis_detected"), .funs = median)


gene.umi.sample 


#nuclei per sample

nucs.sample <- genes.p.nuc.df %>% group_by(sample) %>% summarize(count = n())
nucs.sample

```





### plot genes/nuc

```{r fig.width=6.38, fig.height=3}
genes.p.nuc.plot <-
  ggplot(data = genes.p.nuc.df, aes(x = sample, y = genes_expressed, fill = condition)) +
  
  
 # geom_point(aes(y = genes_expressed, color = condition), 
  #           position = position_jitter(width = 0.2), 
  #             size = 0.1, alpha = 0.1, show.legend = F) +
  
  
  
  rasterise(geom_sina(size = 0.1, alpha = 0.02, aes(color = condition), show.legend = T, scale = F), dpi = 600) +
  
   stat_boxplot(geom ='errorbar') +
  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 1, width = 0.2, color = "black", fill = "white", coef = 0) +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 15),
          
          axis.title.x = element_text(size = 15),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Genes/nucleus") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 4)))

genes.p.nuc.plot

```



### plot nucs per sample


```{r fig.width=6.38, fig.height=3}
nucs.p.sample.plot <-
  ggplot(data = genes.p.nuc.df, aes(x = sample, fill = condition)) +
  
 
  geom_bar(width = 0.7) +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 15),
          
          axis.title.x = element_text(size = 15),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "# of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 4)))

nucs.p.sample.plot


```


## plot doublets


### import data
```{r}
#Import Scz datasets re-mapped with Allen genome and give names to the elements
cms_scz_list_scrub <- lapply(setNames(unlist(samplegroups), unlist(samplegroups)),
                                
                                function(sn){
    Seurat::Read10X_h5(
      paste0(
        "~/projects/human_schizo/MB6-MB23/counts_NON_FILT_GENOME_ENSEMBLE_nova_run1_run2/", sn,
        "/outs/filtered_feature_bc_matrix.h5"
        )
    )
  }
  )
  



#rename nuclei names in matrix, prepend with sample name intead of "one"

cms_scz_list_scrub <- lapply(setNames(unlist(samplegroups), unlist(samplegroups)),
     function(sn) {
       paste0(sn, "_", colnames(cms_scz_list_scrub[[sn]])) %>%
         set_colnames(cms_scz_list_scrub[[sn]], .)
                         
      
      
     }
   )


#amount of nuclei in unfiltered data

cms_scz_list_scrub %>% lapply(function(x) dim(x)[2]) %>% unlist %>% sum




#Filter out cells not present in scrubletf

cms_scz_list_scrub %<>% lapply(function(smpl) {
  smpl[, colnames(smpl) %in% names(scrubletf)]
    }
  )
```



### run conos

```{r}
# pagoda list
p2.list.scrub <- lapply(cms_scz_list_scrub, basicP2proc, n.cores = 70, n.odgenes = 3000, min.cells.per.gene = 0,
                  min.transcripts.per.cell = 0, get.largevis = F, get.tsne = T, make.geneknn = F
)


#conos

con.scrub <- Conos$new(p2.list.scrub,n.cores = 70)


#knn

con.scrub$buildGraph(k = 15, k.self = 5, 
               space = "PCA",
               ncomps = 30, n.odgenes = 3000, matching.method = "mNN", metric = "angular",
               score.component.variance = T, verbose = T,

               )


#UMAP

  con.scrub$embedGraph(method = "UMAP",n.cores = 70, min.dist = 0.1, spread = 15)

```



### Plot scrublet

```{r fig.height=6.02, fig.width=6.02}


#create gg object for getting cluster annotations

gg_annot <-     

    con.scrub$plotGraph(groups = annotations_scz2$med)





doublet.plot <-

con.scrub$plotGraph(alpha = 0.5, size = 0.05, colors = scrubletf, 
                    
                    show.legend = T, font.size = 4, raster = T, raster.height = 12, raster.width = 12
                                        ) +
  #get cluster annotations 
  gg_annot$layers[[2]] +
  scale_size_continuous(range = c(4, 4), guide = F, trans='identity') +
  
 
  theme_void() +
  labs(subtitle = "Doublet score", color = "Score") +
  theme(text = element_text(family = "Liberation Sans", size = 15),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position = c(1, 0.9))
  
  

 doublet.plot 



```
### Excluded doublets plot


```{r fig.height=6.02, fig.width=6.02}

 

doublet.plot.excluded <-

con.scrub$plotGraph(alpha = 0.5, size = 0.05, groups = 
                      
                      
                      setNames(cut(scrubletf, br = c(0, 0.25, 1), include.lowest = T, labels = 
                                     c("Keep", "Exclude")), names(scrubletf)), 
                    
                    show.legend = F, mark.groups = F, font.size = 4, raster = T, raster.height = 12, raster.width = 12
                                        ) +
  #get cluster annotations 
  gg_annot$layers[[2]] +
  scale_size_continuous(range = c(4, 4), guide = F, trans='identity') +
  
 
  theme_void() +
  labs(subtitle = "Doublet exclusion") +
  theme(text = element_text(family = "Liberation Sans", size = 15),
        plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"), legend.position = "top",
        legend.title = element_blank()) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 3))) +
  scale_color_manual(values = c("#779876", "magenta"))
  
  

 doublet.plot.excluded



```

### Figure snRNAseq Technical Plots

```{r fig.width=12.76,fig.height=7.8}
fig.S1 <-
  
      (genes.p.nuc.plot + nucs.p.sample.plot + theme(plot.margin = unit(c(0, 0, 0 ,1), "cm"))) /
        
      (doublet.plot + theme(plot.margin = unit(c(0.5, 3, 0 ,0), "cm")) + 
         
         doublet.plot.excluded + theme(plot.margin = unit(c(0.5, 0, 0 ,1), "cm"))) + 
  
  
  plot_layout(heights = c(1, 3.5))

fig.S1 

```



```{r}
ggsave("~/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/figS1.pdf", fig.S1, width = 12.76, height = 7.8, 
       dpi = 600, device = cairo_pdf)
```





### plot genes/nuc per subtype

```{r fig.width=12.76, fig.height=6}
genes.p.nuc.subt.plot <-
  ggplot(data = genes.p.nuc.df, aes(x = subtype, y = genes_expressed#, fill = subtype
                                    
                                    )) +
  
  
 
  rasterise(geom_sina(size = 0.1, alpha = 0.2, aes(color = subtype), show.legend = F, scale = F), 
            dpi = 600) +
  
  
  stat_boxplot(geom ='errorbar') +
  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 1, width = 0.3, color = "black", fill = "white", coef = 0
               ) +
  
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
         
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
         
          
         
         
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Genes per\nnucleus") +
    scale_color_manual(values = high.pal[levels(genes.p.nuc.df$subtype)])

genes.p.nuc.subt.plot



```

### plot umis/nuc per subtype

```{r fig.width=12.76, fig.height=6}
umi.p.nuc.subt.plot <-
  ggplot(data = genes.p.nuc.df, aes(x = subtype, y = umis_detected
                                    
                                    )) +
  
  
 
  rasterise(geom_sina(size = 0.1, alpha = 0.2, aes(color = subtype), show.legend = F, scale = F), 
            dpi = 600) +
  
  
  stat_boxplot(geom ='errorbar') +
  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 1, width = 0.3, color = "black", fill = "white", coef = 0
               ) +
  
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
         
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
         
          
         
         
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "UMIs per\nnucleus") +
    scale_color_manual(values = high.pal[levels(genes.p.nuc.df$subtype)])
    

umi.p.nuc.subt.plot



```



### plot samples per subtype

```{r fig.width=8.43, fig.height=6}
sampl.p.subt.plot <-
  ggplot(data = samp.p.subt.df, aes(x = subtype, y = freq, fill = subtype)) +

  
  geom_col(show.legend = F, width = 0.9) +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          
          axis.title.x = element_text(size = 15),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
                    text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Samples\nwith ≥ 5 nuclei") +
    scale_color_manual(values = high.pal[levels(genes.p.nuc.df$subtype)]) +
    scale_fill_manual(values = high.pal[levels(genes.p.nuc.df$subtype)]) +
  scale_y_continuous(labels = percent, expand = c(0, 0))

sampl.p.subt.plot

```
### plot subtypes per sample

```{r fig.width=4.33, fig.height=6}
sub.p.sample.plot <-
  ggplot(data = subs.p.sample.df, aes(x = sample, y = freq, fill = condition)) +

  
  geom_col(show.legend = T, width = 0.9) +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          
          axis.title.x = element_text(size = 15),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
         legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
                    text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Subtypes\nwith ≥ 5 nuclei") +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_continuous(labels = percent, expand = c(0, 0))

sub.p.sample.plot

```




### plot nucs in subtypes per sample

```{r fig.width=6.38, fig.height=6}

nuc.p.sample.plot <-
  ggplot(data = nuc.p.sample.df, aes(x = subtype, y = fraction, fill = sample)) +

  
  geom_col(show.legend = T, width = 0.9, position = "stack") +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          
          axis.title.x = element_text(size = 15),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
         legend.title = element_text(size = 15),
          legend.text = element_text(size = 15),
          legend.position = "top",
                    text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_fill_manual(values = cols25()) +
  
  #scale_fill_manual(values = sample.pal2[nuc.p.sample.df$sample %>% unique]) +
  #  scale_color_manual(values = sample.pal2[nuc.p.sample.df$sample %>% unique]) +
  scale_y_continuous(labels = percent, expand = c(0, 0)) +
  guides(fill = guide_legend(nrow = 2))

nuc.p.sample.plot

```
### plot nucs in subtypes per condition

```{r fig.width=6.38, fig.height=6}
nuc.p.cond.plot <-
  ggplot(data = nuc.p.condition.df, aes(x = subtype, y = fraction, fill = condition)) +

  
  geom_col(show.legend = T, width = 0.9, position = "stack") +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 11),
          
          axis.title.x = element_text(size = 15),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
         legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
                    text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_continuous(labels = percent, expand = c(0, 0))

nuc.p.cond.plot

```


### figS snRNAseq Technical Plots

```{r fig.width=12.76,fig.height=15.59}
fig.S3 <-
  
      (genes.p.nuc.subt.plot /
        
      umi.p.nuc.subt.plot /
        
      sampl.p.subt.plot / 
      
      (sub.p.sample.plot + nuc.p.cond.plot + plot_layout(width = c(19, 37))) /
        
      nuc.p.sample.plot) + plot_annotation(tag_levels = "A") &
  theme(plot.tag = element_text(family = "Liberation Sans", size = 18, face = "bold"))

fig.S3 

```

```{r}
ggsave("~/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/figS3.pdf", fig.S3, width = 12.76, height = 15.59, 
       dpi = 600, device = cairo_pdf)
```










# Heatmap Fig 1

for this heatmap no need to remove MB8




## Subset cells for plots

```{r}

set.seed(0)

sampled.cells.high <- unlist(tapply(names(annotations_scz2$high.clean), annotations_scz2$high.clean,
                               function(x) sample(x,min(length(x),1e3))))



sampled.cells.med <- unlist(tapply(names(annotations_scz2$med.clean), annotations_scz2$med.clean,
                               function(x) sample(x,min(length(x),1e3))))

```



## Diff expression
```{r}


#run diff expression on clean no other annotation; MB8 is kept here
de.high.clean <- con_scz$getDifferentialGenes(groups = annotations_scz2$high.clean[sampled.cells.high], n.cores = 70, 
                                              append.auc = T, z.threshold = 0, upregulated.only = T)



de.med.clean <- con_scz$getDifferentialGenes(groups = annotations_scz2$med.clean[sampled.cells.med], n.cores = 70, 
                                              append.auc = T, z.threshold = 0, upregulated.only = T)



```








## Heatmap High

```{r fig.width=10, fig.heigth=20}
#genes to show
genes <- c("CUX2", "LAMP5", "PDGFD", "MARCH1", "FREM3", "SV2C", "UNC5D", "PRSS12", "RORB", "SCHLAP1", "MME", "MET", 
           "ARHGAP15", "THEMIS", "SEMA3A", "NTNG2", "FEZF2", "LRRK1", "TLE4", "HTR2C", "ABO", "SCUBE1", "ADRA1A", 
           "ID2", "LAMP5", "NOS1", "NMBR", "CRH", "NCKAP5", "PAX6", "VIP", "RELN", "ABI3BP", "TYR", "SEMA3C", "SSTR1", 
           "CRH", "SST", "TAC3", "STK32A", "NOS1", "CALB1", "NPY", "MEPE", "SLC1A3", "GFAP", "MBP", "MOG")


genes.more <- c("CUX2", "LAMP5", "PDGFD", "MARCH1", "FREM3", "SV2C", "UNC5D", "PRSS12", "RORB", "SCHLAP1", 
"MME", "MET", "ARHGAP15", "THEMIS", "SEMA3A", "NTNG2", "DCSTAMP", "FEZF2", "LRRK1", "TLE4", "HTR2C", "IL26", 
"CARD11", "KCNIP1", "ABO", "SCUBE1", "CCN2", "ADRA1A", "ID2", "LAMP5", "NOS1", "NMBR", "COL5A2", "NCKAP5", 
"PAX6", "VIP", "RELN", "ABI3BP", "HS3ST3A1", "TYR", "SEMA6A", "TOX2", "SEMA3C", "ADAMTSL1", "PENK", "SSTR1", "OPRM1", 
"CRH", "SST", "TAC3", "STK32A", "NOS1", "CALB1", "HPGD", "NPY", "NOG", "MEPE", "SLC1A3", "MBP", "MOG", 
"PVALB", "CALB2", "TH")


#plot
plot.de <- plotDEheatmap(con = con_scz, groups = annotations_scz2$high.clean[sampled.cells.high] %>% 
                           factor(levels = hsub_order[- length(hsub_order)]),
                         
                         column.metadata = list(samples = con_scz$getDatasetPerCell(), condition = celldiseasef),
                         column.metadata.colors = list(clusters = high.pal, 
                                condition = setNames(c('dodgerblue1', 'indianred1'), c("control", "schizo")),
                                samples = sample.pal),
                         
                         de = de.high.clean, n.genes.per.cluster = 10, 
              show.gene.clusters = T, 
            #  order.clusters = T, 
              use_raster = T,
              show.cluster.legend = T,
              
              raster_device = "CairoPNG",
              additional.genes = genes.more, labeled.gene.subset = genes.more, 
              
              
              #  heatmap_legend_param = list(
             #   
            #    condition = list(labels_gp = grid::gpar(fontsize = 8)),
            #    samples = list(labels_gp = grid::gpar(fontsize = 8)),
             #   
            #    clusters = list()
            #         
            #         
            #         ),
            #  row.label.font.size = 8,

              
            
              raster_quality = 2
              )




#save plot
pdf(file="./plots/plot.de.pdf",width=7.08, height=14.76); print(plot.de); dev.off()


plot.de






annotations_scz2$high.clean[sampled.cells.high] %>% 
                           factor(levels = hsub_order[- length(hsub_order)])
```


## Subset INHIBITORY cells for plots

```{r}


inh.cells <- annotations_scz2$high.clean[

annotations_scz2$high.clean %in% grep("^ID2|^PVAL|^SST|^VIP", levels(annotations_scz2$high.clean), value = T)

] %>% droplevels





set.seed(0)

sampled.cells.high.inh <- unlist(tapply(names(inh.cells), inh.cells,
                               function(x) sample(x,min(length(x),1e3))))




#Only for ID2 and VIP subtypes

inh.cells.small <- annotations_scz2$high.clean[

annotations_scz2$high.clean %in% grep("^ID2|^VIP", levels(annotations_scz2$high.clean), value = T)

] %>% droplevels





set.seed(0)

sampled.cells.high.inh.small <- unlist(tapply(names(inh.cells.small), inh.cells.small,
                               function(x) sample(x,min(length(x),1e3))))







```



## Diff expr INH HIGH
```{r}


#run diff expression on clean no other annotation; MB8 is kept here
de.high.inh.clean <- con_scz$getDifferentialGenes(groups = inh.cells[sampled.cells.high.inh], 
                                                  n.cores = 70, 
                                              append.auc = T, z.threshold = 0, upregulated.only = T)



#repeat for ID2 and VIP only

de.high.inh.clean.small <- con_scz$getDifferentialGenes(groups = inh.cells.small[sampled.cells.high.inh.small], 
                                                  n.cores = 70, 
                                              append.auc = T, z.threshold = 0, upregulated.only = T)




```








## Heatmap INH High

```{r fig.width=10, fig.heigth=12}
#genes to show
inhib_genes2 <- c("GAD2", "SNCG", "ID2", "LAMP5", "NOS1", "NMBR", "COL5A2", "NCKAP5", 
"PAX6", "VIP", "RELN", "ABI3BP", "HS3ST3A1", "TYR", "SEMA6A", "TOX2", "SEMA3C", "ADAMTSL1", "PENK", "SSTR1", "OPRM1", 
"CRH", "PVALB", "MEPE", "SST", "TAC3", "STK32A", "CALB1", "HPGD", "NPY", "NOG", "CALB2", "TH")


#plot
plot.de.inhib <- plotDEheatmap(con = con_scz, groups = inh.cells[sampled.cells.high.inh],
                         
                         column.metadata = list(samples = con_scz$getDatasetPerCell(), condition = celldiseasef),
                         column.metadata.colors = list(clusters = high.pal, 
                                condition = setNames(c('dodgerblue1', 'indianred1'), c("control", "schizo")),
                                samples = sample.pal),
                         
                         de = de.high.clean, n.genes.per.cluster = 5, 
              show.gene.clusters = T, 
            #  order.clusters = T, 
              use_raster = T,
              show.cluster.legend = T,
              
              raster_device = "CairoPNG",
              additional.genes = inhib_genes2, labeled.gene.subset = inhib_genes2, 
              
              raster_quality = 2
              )




#save plot
pdf(file = "./plots/plot.de.inh.pdf",width=10, height=5.7); print(plot.de.inhib); dev.off()


plot.de.inhib



```




## Heatmap INH High SMALLER

```{r fig.width=10, fig.heigth=12}
#genes to show
inhib_genes3 <- c("ID2", "VIP", "CRH", "CALB2")


#plot
plot.de.inhib2 <- plotDEheatmap(con = con_scz, groups = inh.cells[sampled.cells.high.inh],
                         
                         column.metadata = list(samples = con_scz$getDatasetPerCell(), condition = celldiseasef),
                         column.metadata.colors = list(clusters = high.pal, 
                                condition = setNames(c('dodgerblue1', 'indianred1'), c("control", "schizo")),
                                samples = sample.pal),
                         
                         de = de.high.clean, n.genes.per.cluster = 0, 
             # show.gene.clusters = T, 
            #  order.clusters = T, 
              use_raster = T,
              show.cluster.legend = T,
              
              raster_device = "CairoPNG",
              additional.genes = inhib_genes3, labeled.gene.subset = inhib_genes3, 
              
              raster_quality = 2
              )




#save plot
pdf(file = "./plots/plot.de.inh2.pdf",width=10, height=2); print(plot.de.inhib2); dev.off()


plot.de.inhib2



```



## Heatmap INH High EVEN SMALLER

```{r fig.width=10, fig.heigth=12}
#genes to show
inhib_genes3 <- c("ID2", "VIP", "CRH", "CALB2")


#plot
plot.de.inhib3 <- plotDEheatmap(con = con_scz, groups = inh.cells.small[sampled.cells.high.inh.small],
                         
                         column.metadata = list(#samples = con_scz$getDatasetPerCell(), 
                                                condition = celldiseasef),
                         column.metadata.colors = list(clusters = high.pal, 
                                condition = setNames(c('dodgerblue1', 'indianred1'), c("control", "schizo"))#,
                                #samples = sample.pal
                              ),
                         
                         de = de.high.inh.clean.small, n.genes.per.cluster = 0, 
              show.gene.clusters = T, 
            #  order.clusters = T, 
              use_raster = T,
              show.cluster.legend = T,
              
              raster_device = "CairoPNG",
              additional.genes = inhib_genes3, labeled.gene.subset = inhib_genes3, 
              
              raster_quality = 2
              )




#save plot
pdf(file = "./plots/plot.de.inh3.pdf",width=5, height=2.9); print(plot.de.inhib3); dev.off()


plot.de.inhib3



```








## try different type of heatmap

### dataframes
```{r}

#define genes

inhib_genes <- c("ID2", "LAMP5", "NOS1", "NMBR", "COL5A2", "NCKAP5", 
"PAX6", "VIP", "RELN", "ABI3BP", "HS3ST3A1", "TYR", "SEMA6A", "TOX2", "SEMA3C", "ADAMTSL1", "PENK", "SSTR1", "OPRM1", 
"CRH", "PVALB", "MEPE", "SST", "TAC3", "STK32A", "CALB1", "HPGD", "NPY", "NOG", "CALB2", "TH", 
"CHRFAM7A", "CACNA1I", "CYP3A4", "CYP2D6", "CYP2C9")



#create data.frame with those genes

df.inhib.genes <- scz_cm_joint[inhib_genes, ] %>% as.matrix() %>% t() %>% data.frame(check.names = F)


#Add cluster annotations to data frame

df.inhib.genes <- 
  data.frame(df.inhib.genes, subtypes = 
                        annotations_scz2_no_MB8$high[row.names(df.inhib.genes)] %>% 
                        factor(levels = hsub_order), check.names = F
                      
                      )



#Filter, keep only inhibitory


df.inhib.genes <- df.inhib.genes[grepl("^ID2|^PVAL|^SST|^VIP", df.inhib.genes$subtypes), ]

df.inhib.genes$subtypes %<>% droplevels




#mean expression values dataframe

df.inhib.genes.mean <-
  df.inhib.genes %>% group_by(., subtypes) %>% summarize_at(inhib_genes, mean)


above_zero <- function(x) {
  ((x > 0) %>% sum) * 100 / length(x)
  
}


df.inhib.genes.expressed <-
  df.inhib.genes %>% group_by(., subtypes) %>% summarize_at(inhib_genes, above_zero)


#re-check if counting makes sense
#test <- df.inhib.genes[df.inhib.genes$subtypes == "VIP_SSTR1", ]

#(test$CALB2 > 0) %>% sum

df.inhib.genes.mean.melt <-
  melt(df.inhib.genes.mean, id.vars = "subtypes", variable.name = "genes", value.name = "Mean\nExpression")


df.inhib.genes.expressed.melt <-
  melt(df.inhib.genes.expressed, id.vars = "subtypes", variable.name = "genes", value.name = "% Expressed")




df.inhib.mean.expr <- 
      cbind(df.inhib.genes.mean.melt,
        
            `% Expressed` = setNames(df.inhib.genes.expressed.melt$`% Expressed`,
                     paste0(df.inhib.genes.expressed.melt$subtypes, "--", df.inhib.genes.expressed.melt$genes))[
                       
                       paste0(df.inhib.genes.mean.melt$subtypes, "--", df.inhib.genes.mean.melt$genes)]
        
        )



df.inhib.mean.expr$genes %<>% factor(.,
      levels(df.inhib.mean.expr$genes) %>% rev)





```





### plots


```{r, fig.width = 7, fig.height = 8}

plot.inh.markers <- ggplot(data = df.inhib.mean.expr, aes(x = subtypes, y = genes)) + 
  
  geom_point(aes(color = `Mean\nExpression`, size = `% Expressed`)) +
  
  theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, size = 14), 
        plot.title = element_text(hjust = 0.5, size = 18, face = "bold"),
        axis.text.y = element_text(size = 14, face = "italic"), axis.title = element_text(size = 14),
        text = element_text(family = "Helvetica"), axis.title.x = element_blank(), axis.title.y = element_blank(),
        legend.text = element_text(size = 14), legend.title = element_text(size = 14)
        ) +
  
  scale_color_gradient(low =  "#d9d9e1", high = "#1000d8") +
  labs(title = "Inhibitory subtypes markers") +
  scale_size_continuous(range = c(0, 6))
  
#cac9d9
plot.inh.markers

ggsave(filename = "/home/mbatiuk/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/plot.inh.markers.pdf",
        plot.inh.markers, width = 7, height = 8, units = "in", dpi = 600)

```



# Cell fractions


### Prepare data frames with summarized cluster data
```{r}


#data frame with cell names, conditions, and cluster names


df.cellfrac.high <- data.frame(cell_name = names(celldiseasef), condition = celldiseasef, 
                          high_clust = factor(annotations_scz2$high.clean[names(con_scz$getDatasetPerCell())], levels =
                                                hsub_order),
                          sample = con_scz$getDatasetPerCell()
)
  



df.cellfrac.med <- data.frame(cell_name = names(celldiseasef), condition = celldiseasef, 
                          med_clust = factor(annotations_scz2$med.clean[names(con_scz$getDatasetPerCell())], levels =
                                                msub_order),
                          sample = con_scz$getDatasetPerCell()
)
  





#NA come from extra scrublet filter


#Remove NAs
df.cellfrac.high <- df.cellfrac.high[!is.na(df.cellfrac.high$high_clust), ]

df.cellfrac.med <- df.cellfrac.med[!is.na(df.cellfrac.med$med_clust), ]


#Rename MB8 as MB8-2 - same sample in the end:

df.cellfrac.med$sample %<>% gsub("^MB8$", "MB8-2", .)
df.cellfrac.high$sample %<>% gsub("^MB8$", "MB8-2", .)

#Remove glia - CONTAMINATION DURING SORT - DISTRIBUTION STOCHASTIC BASED ON SORT GATES

df.cellfrac.med <- df.cellfrac.med[!(df.cellfrac.med$med_clust == "Glia"), ]
df.cellfrac.high <- df.cellfrac.high[!(df.cellfrac.high$high_clust == "Glia"), ]

df.cellfrac.high$high_clust %<>% droplevels
df.cellfrac.med$med_clust %<>% droplevels






#Summarize data

df.cellrac.summarized.high <- df.cellfrac.high %>% group_by(high_clust, sample) %>% summarize(n =  n()) %>% group_by(sample) %>% mutate(freq = n / sum(n))
df.cellrac.summarized.med <- df.cellfrac.med %>% group_by(med_clust, sample) %>% summarize(n =  n()) %>% group_by(sample) %>% mutate(freq = n / sum(n))

df.cellrac.summarized.med$condition <- diseasef[df.cellrac.summarized.med$sample]
df.cellrac.summarized.high$condition <- diseasef[df.cellrac.summarized.high$sample]



#Log-transform - explore possibility of t-test
df.cellrac.summarized.med %<>% mutate(freq_log = log1p(freq))
df.cellrac.summarized.high %<>% mutate(freq_log = log1p(freq))

#log(max(x+1) - x) transformation
df.cellrac.summarized.med %<>% mutate(freq_logmax = log(max(freq + 1) - freq))
df.cellrac.summarized.high %<>% mutate(freq_logmax = log(max(freq + 1) - freq))

#sqrt transform
df.cellrac.summarized.med %<>% mutate(freq_sqrt = sqrt(freq))
df.cellrac.summarized.high %<>% mutate(freq_sqrt = sqrt(freq))

#sqrt(max(x+1) - x) transform

df.cellrac.summarized.med %<>% mutate(freq_sqrtmax = sqrt(max(freq+1) - freq))
df.cellrac.summarized.high %<>% mutate(freq_sqrtmax = sqrt(max(freq+1) - freq))


#reciprocal transform
df.cellrac.summarized.med %<>% mutate(freq_recip = 1/freq)
df.cellrac.summarized.high %<>% mutate(freq_recip = 1/freq)

#1/(max(x+1) - x) reciprocal transform
df.cellrac.summarized.med %<>% mutate(freq_recipmax = 1/(max(freq+1) - freq))
df.cellrac.summarized.high %<>% mutate(freq_recipmax = 1/(max(freq+1) - freq))




```




## Statistical tests

### Normality, Shapiro-Wilk

```{r}


       
#Medium resolution

df.cellrac.summarized.med.list <- 
    split(df.cellrac.summarized.med, f = df.cellrac.summarized.med$condition) %>% lapply(., function(x) {
      split(x, f = x$med_clust)
      }
      ) %>% unlist(recursive = F)




shapiro.cellfraq.med <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq))

shapiro.cellfraq.med

#High resolution

df.cellrac.summarized.high.list <- 
    split(df.cellrac.summarized.high, f = df.cellrac.summarized.high$condition) %>% lapply(., function(x) {
      split(x, f = x$high_clust)
      }
      ) %>% unlist(recursive = F)




shapiro.cellfraq.high <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq))

shapiro.cellfraq.high


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```

Medium resolution: Majority have normal distr, but some not - better to stick to non-parametrics
High resolution: Majority have normal distr, but some not - better to stick to non-parametrics


### Normality, Shapiro-Wilk no condition split

```{r}


       
#Medium resolution

df.cellrac.summarized.med.list.nosplit <- 
    split(df.cellrac.summarized.med, f = df.cellrac.summarized.med$med_clust)




shapiro.cellfraq.med.nosplit <-
  lapply(df.cellrac.summarized.med.list.nosplit, function(x) shapiro.test(x$freq))

shapiro.cellfraq.med.nosplit

#High resolution

df.cellrac.summarized.high.list.nosplit <- 
    split(df.cellrac.summarized.high, f = df.cellrac.summarized.high$high_clust)




shapiro.cellfraq.high.nosplit <-
  lapply(df.cellrac.summarized.high.list.nosplit, function(x) shapiro.test(x$freq))

shapiro.cellfraq.high.nosplit


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
some non-normal




### Normality, Shapiro-Wilk log-transformed

```{r}


       
#Medium resolution

shapiro.cellfraq.med.log <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq_log))

shapiro.cellfraq.med.log

#High resolution


shapiro.cellfraq.high.log <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq_log))

shapiro.cellfraq.high.log


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```

log1p - non-normal for some


### Normality, Shapiro-Wilk log-transformed no condition split

```{r}


       
#Medium resolution


shapiro.cellfraq.med.nosplit.log <-
  lapply(df.cellrac.summarized.med.list.nosplit, function(x) shapiro.test(x$freq_log))

shapiro.cellfraq.med.nosplit.log

#High resolution


shapiro.cellfraq.high.nosplit.log <-
  lapply(df.cellrac.summarized.high.list.nosplit, function(x) shapiro.test(x$freq_log))

shapiro.cellfraq.high.nosplit.log


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```


### Normality, Shapiro-Wilk log(max(x+1) - x)-transformed

```{r}


       
#Medium resolution

shapiro.cellfraq.med.logmax <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq_logmax))

shapiro.cellfraq.med.logmax

#High resolution


shapiro.cellfraq.high.logmax <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq_logmax))

shapiro.cellfraq.high.logmax


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
non-normal for some


### Normality, Shapiro-Wilk sqrt-transformed

```{r}


       
#Medium resolution

shapiro.cellfraq.med.sqrt <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq_sqrt))

shapiro.cellfraq.med.sqrt

#High resolution


shapiro.cellfraq.high.sqrt <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq_sqrt))

shapiro.cellfraq.high.sqrt


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
sqrt - non-normal for some


### Normality, Shapiro-Wilk log-transformed no condition split

```{r}


       
#Medium resolution


shapiro.cellfraq.med.nosplit.sqrt <-
  lapply(df.cellrac.summarized.med.list.nosplit, function(x) shapiro.test(x$freq_sqrt))

shapiro.cellfraq.med.nosplit.sqrt

#High resolution


shapiro.cellfraq.high.nosplit.sqrt <-
  lapply(df.cellrac.summarized.high.list.nosplit, function(x) shapiro.test(x$freq_sqrt))

shapiro.cellfraq.high.nosplit.sqrt


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
some non-normal


### Normality, Shapiro-Wilk sqrt(max(x+1) - x)-transformed

```{r}


       
#Medium resolution

shapiro.cellfraq.med.sqrtmax <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq_sqrtmax))

shapiro.cellfraq.med.sqrtmax

#High resolution


shapiro.cellfraq.high.sqrtmax <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq_sqrtmax))

shapiro.cellfraq.high.sqrtmax


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
sqrt(max(x+1) - x) - non-normal for some



### Normality, Shapiro-Wilk reciprocal-transformed

```{r}


       
#Medium resolution

shapiro.cellfraq.med.recip <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq_recip))

shapiro.cellfraq.med.recip

#High resolution


shapiro.cellfraq.high.recip <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq_recip))

shapiro.cellfraq.high.recip


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
reciprocal - non-normal for some

### Normality, Shapiro-Wilk reciprocal-transformed no condition split

```{r}


       
#Medium resolution


shapiro.cellfraq.med.nosplit.recip <-
  lapply(df.cellrac.summarized.med.list.nosplit, function(x) shapiro.test(x$freq_recip))

shapiro.cellfraq.med.nosplit.recip

#High resolution


shapiro.cellfraq.high.nosplit.recip <-
  lapply(df.cellrac.summarized.high.list.nosplit, function(x) shapiro.test(x$freq_recip))

shapiro.cellfraq.high.nosplit.recip


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
many non-normal



### Normality, Shapiro-Wilk 1/(max(x+1) - x) reciprocal-transformed

```{r}


       
#Medium resolution

shapiro.cellfraq.med.recipmax <-
  lapply(df.cellrac.summarized.med.list, function(x) shapiro.test(x$freq_recipmax))

shapiro.cellfraq.med.recipmax

#High resolution


shapiro.cellfraq.high.recipmax <-
  lapply(df.cellrac.summarized.high.list, function(x) shapiro.test(x$freq_recipmax))

shapiro.cellfraq.high.recipmax


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
non-normal for some


### Normality, Shapiro-Wilk on box-cox transformed data

```{r}
#Medium resolution




df.cellrac.summarized.med.list.bc <- 
    split(df.cellrac.summarized.med, f = df.cellrac.summarized.med$med_clust)

#perform box-cox transformation

df.cellrac.summarized.med.list.bc <-
    lapply(df.cellrac.summarized.med.list.bc, function(z) {
     bc_result <- boxCox(z$freq ~ z$condition, plotit = F)
    
    opt.lambda <- bc_result$x[which.max(bc_result$y)]
    
    z %<>% mutate(freq_bc = bcPower(freq, lambda = opt.lambda))
    z
    }
    )

#further split by condition:

#df.cellrac.summarized.med.list.bc <- lapply(df.cellrac.summarized.med.list.bc, function(x) {
#    split(x, f = x$condition)
#}
#) %>% unlist(recursive = F)





shapiro.cellfraq.med.bc <-
  lapply(df.cellrac.summarized.med.list.bc, function(x) shapiro.test(x$freq_bc))

shapiro.cellfraq.med.bc

#High resolution

df.cellrac.summarized.high.list.bc <- 
    split(df.cellrac.summarized.high, f = df.cellrac.summarized.high$high_clust)

#perform box-cox transformation

df.cellrac.summarized.high.list.bc <-
    lapply(df.cellrac.summarized.high.list.bc, function(z) {
     bc_result <- boxCox(z$freq ~ z$condition, plotit = F)
    
    opt.lambda <- bc_result$x[which.max(bc_result$y)]
    
    z %<>% mutate(freq_bc = bcPower(freq, lambda = opt.lambda))
    z
    }
    )

#further split by condition:

#df.cellrac.summarized.high.list.bc <- lapply(df.cellrac.summarized.high.list.bc, function(x) {
#    split(x, f = x$condition)
#}
#) %>% unlist(recursive = F)





shapiro.cellfraq.high.bc <-
  lapply(df.cellrac.summarized.high.list.bc, function(x) shapiro.test(x$freq_bc))

shapiro.cellfraq.high.bc


lapply(shapiro.cellfraq.med.bc, function(x) x$p.value) %>% unlist %>% `[`(., . <= 0.05)
lapply(shapiro.cellfraq.high.bc, function(x) x$p.value) %>% unlist %>% `[`(., . <= 0.05)


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```



### Normality, Shapiro-Wilk Yeo-Johnson transformed

```{r}
#Medium resolution




df.cellrac.summarized.med.list.yj <- 
    split(df.cellrac.summarized.med, f = df.cellrac.summarized.med$med_clust)

#perform box-cox transformation

df.cellrac.summarized.med.list.yj <-
    lapply(df.cellrac.summarized.med.list.yj, function(z) {
     bc_result <- boxCox(z$freq ~ z$condition, plotit = F, family = "yjPower")
    
    opt.lambda <- bc_result$x[which.max(bc_result$y)]
    
    z %<>% mutate(freq_yj = yjPower(freq, lambda = opt.lambda))
    z
    }
    )

#further split by condition:

#df.cellrac.summarized.med.list.bc <- lapply(df.cellrac.summarized.med.list.bc, function(x) {
#    split(x, f = x$condition)
#}
#) %>% unlist(recursive = F)





shapiro.cellfraq.med.yj <-
  lapply(df.cellrac.summarized.med.list.yj, function(x) shapiro.test(x$freq_yj))

shapiro.cellfraq.med.yj

#High resolution

df.cellrac.summarized.high.list.yj <- 
    split(df.cellrac.summarized.high, f = df.cellrac.summarized.high$high_clust)

#perform box-cox transformation

df.cellrac.summarized.high.list.yj <-
    lapply(df.cellrac.summarized.high.list.yj, function(z) {
     bc_result <- boxCox(z$freq ~ z$condition, plotit = F, family = "yjPower")
    
    opt.lambda <- bc_result$x[which.max(bc_result$y)]
    
    z %<>% mutate(freq_yj = yjPower(freq, lambda = opt.lambda))
    z
    }
    )

#further split by condition:

#df.cellrac.summarized.high.list.bc <- lapply(df.cellrac.summarized.high.list.bc, function(x) {
#    split(x, f = x$condition)
#}
#) %>% unlist(recursive = F)





shapiro.cellfraq.high.yj <-
  lapply(df.cellrac.summarized.high.list.yj, function(x) shapiro.test(x$freq_yj))

shapiro.cellfraq.high.yj


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```

some non-normal with Yeo_Johnson transformation



### Normality, Shapiro-Wilk on top CoDa subtypes 

```{r}
#Medium resolution




df.cellrac.summarized.med.list.coda <- 
    split(df.cellrac.summarized.med, f = df.cellrac.summarized.med$med_clust) %>% `[`(., 
      
      c("SST", "L2_3_CUX2_FREM3", "L4_5_FEZF2_LRRK1", "L3_CUX2_PRSS12",
           "PVALB")
    )



#further split by condition:

df.cellrac.summarized.med.list.coda <- lapply(df.cellrac.summarized.med.list.coda, function(x) {
    split(x, f = x$condition)
}
) %>% unlist(recursive = F)





shapiro.cellfraq.med.coda <-
  lapply(df.cellrac.summarized.med.list.coda, function(x) shapiro.test(x$freq))

shapiro.cellfraq.med.coda

#High resolution

df.cellrac.summarized.high.list.coda <- 
    split(df.cellrac.summarized.high, f = df.cellrac.summarized.high$high_clust) %>% `[`(., 
      
      c("SST_CALB1", "L2_3_CUX2_FREM3_SV2C", "L4_5_FEZF2_LRRK1", "L3_CUX2_PRSS12",
           "SST_NOS1", "SST_STK32A", "L2_3_CUX2_FREM3_UNC5D", "PVALB_MEPE")
    )



#further split by condition:

df.cellrac.summarized.high.list.coda <- lapply(df.cellrac.summarized.high.list.coda, function(x) {
    split(x, f = x$condition)
}
) %>% unlist(recursive = F)





shapiro.cellfraq.high.coda <-
  lapply(df.cellrac.summarized.high.list.coda, function(x) shapiro.test(x$freq))

shapiro.cellfraq.high.coda


#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```

non-normal for some


### Equality of variances (Levene's test):


```{r}
#Medium resolution
df.cellrac.summarized.med.list2 <- 
    split(df.cellrac.summarized.med, f = df.cellrac.summarized.med$med_clust)


levene.med <- lapply(df.cellrac.summarized.med.list2, function(x) leveneTest(freq ~ condition, data = x))
levene.med




#High resolution
df.cellrac.summarized.high.list2 <- 
    split(df.cellrac.summarized.high, f = df.cellrac.summarized.high$high_clust)


levene.high <- lapply(df.cellrac.summarized.high.list2, function(x) leveneTest(freq ~ condition, data = x))
levene.high

```


Med resolution: all  are homogeneous
High resolution: all  are homogeneous


### Equality of variances (Levene's test) boxcox:

```{r}
#Medium resolution



levene.med.bc <- lapply(df.cellrac.summarized.med.list.bc, function(x) leveneTest(freq_bc ~
                                              condition, data = x))
levene.med.bc




#High resolution



levene.high.bc <- lapply(df.cellrac.summarized.high.list.bc, function(x) leveneTest(freq_bc ~ condition, data = x))
levene.high.bc

```
all homogeneous


### Independent 2-group Mann-Whitney U test
```{r}

wilcox.high <-
  lapply(df.cellrac.summarized.high.list2, function(x) {wilcox.test(freq ~ condition, data = x)})
wilcox.high


wilcox.med <-
  lapply(df.cellrac.summarized.med.list2, function(x) {wilcox.test(freq ~ condition, data = x)})
wilcox.med



lapply(wilcox.med, function(x) x$p.value) %>% unlist %>% `[`(., . <= 0.05)
lapply(wilcox.high, function(x) x$p.value) %>% unlist %>% `[`(., . <= 0.05)


#multiple comparison-adjusted values

wilcox.med.adj <-
lapply(wilcox.med, function(x) x$p.value) %>% unlist %>% 
  p.adjust(method = "BH") 

wilcox.med.adj %>% `[`(., . <= 0.05)

wilcox.high.adj <-
lapply(wilcox.high, function(x) x$p.value) %>% unlist %>% 
  p.adjust(method = "BH") 

wilcox.high.adj %>% `[`(., . <= 0.05)



#Focus on subtypes prioritised by CoDa - take ONLY few highest CoDa hits

wilcox.high.coda.adj <-
    lapply(wilcox.high, function(x) x$p.value) %>% unlist %>% 
  
  `[`(., c("SST_CALB1", "L2_3_CUX2_FREM3_SV2C", "L4_5_FEZF2_LRRK1", "L3_CUX2_PRSS12",
           "SST_NOS1", "SST_STK32A", "L2_3_CUX2_FREM3_UNC5D", "PVALB_MEPE"
           
           )) %>%

      p.adjust(method = "BH")

wilcox.high.coda.adj


wilcox.med.coda.adj <-
    lapply(wilcox.med, function(x) x$p.value) %>% unlist %>% 
  
  `[`(., c("SST", "L2_3_CUX2_FREM3", "L4_5_FEZF2_LRRK1", "L3_CUX2_PRSS12",
           "PVALB"
           
           )) %>%

      p.adjust(method = "BH")

wilcox.med.coda.adj



```







### Signif function - p val to *

```{r}
#Define Significance function for changing p values to *
signif <- function(x) {
  
  y <- x
  y[x <= 0.001] <- "***"
  y[x <= 0.01 & x > 0.001] <- "**"
  y[x <= 0.05 & x > 0.01] <- "*"
  y[x > 0.05] <- "NS"
  y[is.nan(x)] <- NaN
  y[is.na(x) & !is.nan(x)] <- NA
 
  
  
  
  return(y)
}

```



### Plot Distribution medium
```{r fig.width=4.25, fig.height=5}

cell.fraction.plot.med <-
  ggplot(data = df.cellrac.summarized.med, aes(x = med_clust, y = freq, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = freq, color = condition), position = position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.4, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.5) +
  
  
  
  geom_signif(annotations = signif(unlist(lapply(wilcox.med, function(x) x$p.value)))[!unlist(lapply(wilcox.med, function(x) x$p.value)) > 0.05],
             xmin = c(0.75:13.75)[!unlist(lapply(wilcox.med, function(x) x$p.value)) > 0.05], 
             xmax = c(1.25:14.25)[!unlist(lapply(wilcox.med, function(x) x$p.value)) > 0.05], 
             y_position = c(rep(sqrt(max(
                df.cellrac.summarized.med$freq)*0.9), 14))[!unlist(lapply(wilcox.med, 
                                                                                function(x) x$p.value)) > 0.05],
             textsize = 5, #vjust = 0.5
              ) +
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_sqrt(breaks = c(10e-4, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3), limits = c(0.00, max(
          df.cellrac.summarized.med$freq)*1.15), labels = percent) +
  labs(subtitle = "Medium resolution")

cell.fraction.plot.med






```


### Plot Distribution medium adjusted p vals
```{r fig.width=4.25, fig.height=5}

df.cellrac.summarized.med$condition %<>% gsub("^schizo.*", "schizophrenia", .)

cell.fraction.plot.med.adj <-
  ggplot(data = df.cellrac.summarized.med, aes(x = med_clust, y = freq, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = freq, color = condition), position = position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3, lwd = 0.4) +
  
  
  
  geom_signif(annotations = signif(wilcox.med.adj)[!wilcox.med.adj > 0.05],
             xmin = c(0.75:13.75)[!wilcox.med.adj > 0.05], 
             xmax = c(1.25:14.25)[!wilcox.med.adj > 0.05], 
             y_position = c(rep(sqrt(max(
                df.cellrac.summarized.med$freq)*0.9), 14))[!wilcox.med.adj > 0.05],
             textsize = 5, #vjust = 0.5
              ) +
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_sqrt(breaks = c(10e-4, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3), limits = c(0.00, max(
          df.cellrac.summarized.med$freq)*1.15), labels = percent) +
  labs(subtitle = "Medium resolution")

cell.fraction.plot.med.adj






```








### Plot Distribution high
```{r fig.width=8.5, fig.height=5.5}




cell.fraction.plot.high <- 

  ggplot(data = df.cellrac.summarized.high, aes(x = high_clust, y = freq, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = freq, color = condition), position = position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.4, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.5) +
  
  
  
  geom_signif(annotations = signif(unlist(lapply(wilcox.high, function(x) x$p.value)))[!unlist(lapply(wilcox.high, function(x) x$p.value)) > 0.05],
             xmin = c(0.75:34.75)[!unlist(lapply(wilcox.high, function(x) x$p.value)) > 0.05], 
             xmax = c(1.25:35.25)[!unlist(lapply(wilcox.high, function(x) x$p.value)) > 0.05], 
             y_position = c(rep(sqrt(max(
                df.cellrac.summarized.high$freq)*0.9), 35))[!unlist(lapply(wilcox.high, 
                                                                                function(x) x$p.value)) > 0.05],
             textsize = 5#, vjust = 0.5
              ) +
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          
          axis.title.x = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_sqrt(breaks = c(10e-4, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3), limits = c(0.00, max(
          df.cellrac.summarized.med$freq)*1.15), labels = percent) +
  labs(subtitle = "High resolution")


cell.fraction.plot.high





```


### Plot Distribution high - adjusted p-vals
```{r fig.width=8.5, fig.height=5.5}
df.cellrac.summarized.high$condition %<>% gsub("^schizo.*", "schizophrenia", .)

cell.fraction.plot.high.adj <- 

  ggplot(data = df.cellrac.summarized.high, aes(x = high_clust, y = freq, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = freq, color = condition), position = position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3, lwd = 0.4) +
  
  
  
  geom_signif(annotations = signif(wilcox.high.adj)[!wilcox.high.adj > 0.05],
             xmin = c(0.75:34.75)[!wilcox.high.adj > 0.05], 
             xmax = c(1.25:35.25)[!wilcox.high.adj > 0.05], 
             y_position = c(rep(sqrt(max(
                df.cellrac.summarized.high$freq)*0.9), 35))[!wilcox.high.adj > 0.05],
             textsize = 5#, vjust = 0.5
              ) +
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          
          axis.title.x = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_sqrt(breaks = c(10e-4, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3), limits = c(0.00, max(
          df.cellrac.summarized.med$freq)*1.15), labels = percent) +
  labs(subtitle = "High resolution")


cell.fraction.plot.high.adj





```



```{r fig.width=12.76, fig.height=6}
cell.fraction.plot.full.med.high <-

cell.fraction.plot.med + cell.fraction.plot.high +
  plot_layout(guides = 'collect', width = c(14, 35)) & theme(legend.position = "top")

cell.fraction.plot.full.med.high
```

```{r}
ggsave("~/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/figS7A.pdf", cell.fraction.plot.full.med.high,
      device = cairo_pdf, width = 12.76, height = 6, dpi = 600)



```

### Plot distrib med high adjusted p values

```{r fig.width=12.76, fig.height=5.6}
cell.fraction.plot.full.med.high.adj <-

cell.fraction.plot.med.adj + cell.fraction.plot.high.adj +
  plot_layout(guides = 'collect', width = c(14, 35)) & theme(legend.position = "top",
                      legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm"))

cell.fraction.plot.full.med.high.adj
```

```{r}
ggsave("~/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/figS7A_dec_2020.pdf", cell.fraction.plot.full.med.high.adj,
      device = cairo_pdf, width = 12.76, height = 5.6, dpi = 600)



```









### Plot Distribution medium subset NEW
```{r fig.width=2.7, fig.height=4.5}

df.cellrac.summarized.med.sub2 <- df.cellrac.summarized.med[
  
  grepl("^L2_3_CUX2_FREM3$|^L3_CUX2_PRSS12$|^L4_5_FEZF2_LRRK1$|^ID2_LAMP5$|^ID2_NCKAP5$|^VIP$|^PVALB$|^SST$",
                                                                 df.cellrac.summarized.med$med_clust),
                                                             ]


df.cellrac.summarized.med.sub2$condition %<>% gsub("^schizo.*",
                                                  "schizophrenia", .)


wilcox.med.vect.subs <- wilcox.med.adj[grepl(
  "^L2_3_CUX2_FREM3$|^L3_CUX2_PRSS12$|^L4_5_FEZF2_LRRK1$|^ID2_LAMP5$|^ID2_NCKAP5$|^VIP$|^PVALB$|^SST$",
                                              names(wilcox.med.adj))]




cell.fraction.plot.med.sub2 <-
  ggplot(data = df.cellrac.summarized.med.sub2, aes(x = med_clust, y = freq, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = freq, color = condition), position = position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3, lwd = 0.4) +
  
  
  
  geom_signif(annotations = signif(wilcox.med.vect.subs)[!wilcox.med.vect.subs > 0.05],
             xmin = c(0.75:7.75)[!wilcox.med.vect.subs > 0.05], 
             xmax = c(1.25:8.25)[!wilcox.med.vect.subs > 0.05], 
             y_position = c(rep(sqrt(max(
                df.cellrac.summarized.high.sub$freq)*1.3), 8))[!wilcox.med.vect.subs > 0.05],
             textsize = 5, #vjust = 0.5
              ) +
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_sqrt(breaks = c(0.001, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3), limits = c(0.00, max(
          df.cellrac.summarized.high.sub$freq)*1.6), labels = percent) +
#ylim(0, max(
#                df.cellrac.summarized.med.sub$freq)*1.15) +
  labs(subtitle = "Medium")

cell.fraction.plot.med.sub2


```



### Plot Distribution high subset NEW
```{r fig.width=2.1, fig.height=4.5}


df.cellrac.summarized.high.sub2 <- df.cellrac.summarized.high[
  
  grepl("^L2_3_CUX2_FREM3_SV2C$|^L2_3_CUX2_FREM3_UNC5D$|L3_CUX2_PRSS12$|^L4_5_FEZF2_LRRK1$|^L5_6_FEZF2_TLE4_HTR2C$|^ID2_LAMP5_NMBR$|^VIP_CRH$|^PVALB_SST$|^ID2_NCKAP5$|^SST_CALB1$|^SST_STK32A$",
                                                                 df.cellrac.summarized.high$high_clust),
                                                             ]



wilcox.high.vect.subs <- wilcox.high.adj[grepl(
  "^L2_3_CUX2_FREM3_SV2C$|^L2_3_CUX2_FREM3_UNC5D$|L3_CUX2_PRSS12$|^L4_5_FEZF2_LRRK1$|^L5_6_FEZF2_TLE4_HTR2C$|^ID2_LAMP5_NMBR$|^VIP_CRH$|^PVALB_SST$|^ID2_NCKAP5$|^SST_CALB1$|^SST_STK32A$",
                                              names(wilcox.high.adj))]


df.cellrac.summarized.high.sub2$condition %<>% gsub("^schizo.*",
                                                  "schizophrenia", .)

cell.fraction.plot.high.sub2 <-
  ggplot(data = df.cellrac.summarized.high.sub2, aes(
    x = high_clust, y = freq, dodge = condition, fill = condition)) +
  
  
  
    
  
  geom_point(aes(y = freq, color = condition), position = position_jitterdodge(dodge.width = 0.7, jitter.width = 0.5), 
               size = 1.1, alpha = 0.2, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3, lwd = 0.4) +
  
  
  
  geom_signif(annotations = signif(wilcox.high.vect.subs)[!wilcox.high.vect.subs > 0.05],
             xmin = c(0.75:10.75)[!wilcox.high.vect.subs > 0.05], 
             xmax = c(1.25:11.25)[!wilcox.high.vect.subs > 0.05], 
             y_position = c(rep(sqrt(max(
                df.cellrac.summarized.high.sub$freq)*1.3), 11))[!wilcox.high.vect.subs > 0.05],
             textsize = 5, #vjust = 0.5
              ) +
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 12),
          
          axis.title.x = element_blank(),
          axis.text.y = element_blank(),
          axis.title.y = element_blank(),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank()) +
          labs(x = NULL, y = "Fraction of nuclei") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  scale_y_sqrt(breaks = c(10e-4, 0.05, 0.1, 0.15, 0.2, 0.25, 0.3), limits = c(0.00, max(
          df.cellrac.summarized.high.sub$freq)*1.6), labels = percent) +
#ylim(0, max(
#                df.cellrac.summarized.high.sub$freq)*1.15) +
  labs(subtitle = "High")

cell.fraction.plot.high.sub2




```



```{r fig.width=5, fig.height=4.8}
cell.fraction.plot.med.high2 <-


cell.fraction.plot.med.sub2 + cell.fraction.plot.high.sub2 +
  plot_layout(guides = 'collect', width = c(8, 11)) & theme(legend.position = "top",
    legend.margin =  margin(t = -1, b = -1.5, r = 0, l = 0, unit = "mm"))



cell.fraction.plot.med.high2

```

```{r}
ggsave("~/projects/human_schizo/MB6-MB23/2ndary/schizo_notebook_2/plots/fig2B_dec_2020.pdf", cell.fraction.plot.med.high2,
      device = cairo_pdf, width = 5, height = 4.8, dpi = 600)


```





# Covariate barplots

## Normality Shapiro-Wilk

```{r}
meta.list <- 
    list(age = metadata[, c("Identifier", "Age", "Diagnosis")],
         pmi = metadata[, c("Identifier", "PMI_hrs", "Diagnosis")],
         ph = metadata[, c("Identifier", "pH_liquor", "Diagnosis")],
         duration = metadata[, c("Identifier", "Neuroleptic_treatment_duration_years", "Diagnosis")]
         
         
         ) %>% lapply(., function(x) {
      split(x, f = x$Diagnosis)
}) %>% unlist(recursive = F)

#remove MB8

meta.list <-

lapply(meta.list, function(x) {
 x <- x[x$Identifier != "MB8", ]
 x  
})



#remove n.a. ph

meta.list$ph.control <- meta.list$ph.control[meta.list$ph.control$pH_liquor != "n.a.", ]
meta.list$ph.schizo <- meta.list$ph.schizo[meta.list$ph.schizo$pH_liquor != "n.a.", ]

meta.list$ph.control$pH_liquor %<>% as.numeric()
meta.list$ph.schizo$pH_liquor %<>% as.numeric()


#remove n.a. neuroleptic duration
meta.list$duration.control <-
  meta.list$duration.control[
    meta.list$duration.control$Neuroleptic_treatment_duration_years != "n.a.",
  ]
  
  
  
meta.list$duration.schizo <-
  meta.list$duration.schizo[
    meta.list$duration.schizo$Neuroleptic_treatment_duration_years != "n.a.",
  ]

meta.list$duration.control$Neuroleptic_treatment_duration_years %<>% as.numeric()
meta.list$duration.schizo$Neuroleptic_treatment_duration_years %<>% as.numeric()




#shapiro test

shapiro.meta <-
  lapply(meta.list, function(x) {tryCatch(shapiro.test(x[, 2]),
                                                                  error = function(e) NULL)
    
    })






shapiro.meta

#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```

pmi.control - non-normal distribution
duration.control - non-normal distribution


## Equality of variances (Levene's test):

```{r}
#Medium resolution
meta.list2 <- 
    list(age = metadata[, c("Identifier", "Age", "Diagnosis")],
         pmi = metadata[, c("Identifier", "PMI_hrs", "Diagnosis")],
         ph = metadata[, c("Identifier", "pH_liquor", "Diagnosis")],
         duration = metadata[, c("Identifier", "Neuroleptic_treatment_duration_years", "Diagnosis")]
         
         )

#remove MB8

meta.list2 <-

lapply(meta.list2, function(x) {
 x <- x[x$Identifier != "MB8", ]
 x  
})




#RENAME schizo to SZ
meta.list2 <-

lapply(meta.list2, function(x) {
  
  x$Diagnosis <- x$Diagnosis %>% gsub("schizo", "SZ", .)
  x
  
})




#remove n.a., fix numeric values

meta.list2$ph <- meta.list2$ph[meta.list2$ph$pH_liquor != "n.a.", ]

meta.list2$ph$pH_liquor %<>% as.numeric()


meta.list2$duration <- meta.list2$duration[meta.list2$duration$Neuroleptic_treatment_duration_years != "n.a.", ]
meta.list2$duration$Neuroleptic_treatment_duration_years %<>% as.numeric()



levene.meta <- lapply(meta.list2, function(x) leveneTest(x[, 2] ~ x$Diagnosis))


levene.meta.p <-
lapply(levene.meta, function(x) {
  y <- x$`Pr(>F)`[1]
  #return(y)
  
}) 


 unlist(levene.meta.p) <= 0.05
 
 
 #p < 0.05 - unequal variances
```

variances are equal for all except duration of treatment



## Independent 2-group Mann-Whitney U test
```{r}

wilcox.meta <-
  lapply(meta.list2, function(x) {wilcox.test(x[, 2] ~ x$Diagnosis)})


wilcox.meta




```   
not significant

## T-test - equal variances (all except duration of treatments)

```{r}



ttest.meta <-
  lapply(meta.list2, function(x) {t.test(x[, 2] ~ x$Diagnosis, var.equal = T)})


ttest.meta




``` 

not significant




## T-test - non-equal variances

```{r}



ttest.meta.noneq <-
  lapply(meta.list2, function(x) {t.test(x[, 2] ~ x$Diagnosis, var.equal = F)})


ttest.meta.noneq




``` 





```{r fig.width=2, fig.height=3}
age.plot <-
  ggplot(data = meta.list2$age, aes(x = 1, y = Age, dodge = Diagnosis, fill = Diagnosis)) +
  
  
  
    
  
  
  
geom_point(aes(y = Age, color = Diagnosis), position = position_jitterdodge(dodge.width = 2.3, jitter.width = 0.2), 
               size = 1.1, alpha = 0.5, show.legend = F) +
  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3) +
  
  
  
  geom_signif(annotations = signif(ttest.meta$age$p.value),
             xmin = 0.8, 
             xmax = 1.2, 
             y_position = max(meta.list2$age$Age)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
  
  stat_summary(fun = "mean", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.75), 
                 mapping = aes(group = Diagnosis), fill = "grey",
                 shape = 23
           ) + 
  
  
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          axis.ticks.x = element_blank(),
          panel.grid.major.x = element_blank()
          ) +
          labs(x = NULL, y = "Age, years") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
 
ylim(0, max(
                meta.list2$age$Age)*1.15)
 

age.plot



```


```{r fig.width=2, fig.height=3}
pmi.plot <-
  ggplot(data = meta.list2$pmi, aes(x = 1, y = PMI_hrs, dodge = Diagnosis, fill = Diagnosis)) +
  
  
   geom_point(aes(y = PMI_hrs, color = Diagnosis), 
             position = position_jitterdodge(dodge.width = 2.3, jitter.width = 0.2), 
               size = 1.1, alpha = 0.5, show.legend = F) +
    
  
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3) +
  
  
  
  geom_signif(annotations = signif(ttest.meta$pmi$p.value),
             xmin = 0.8, 
             xmax = 1.2, 
             y_position = max(meta.list2$pmi$PMI_hrs)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
   stat_summary(fun = "mean", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.75), 
                 mapping = aes(group = Diagnosis), fill = "grey",
                 shape = 23
           ) + 
 
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          axis.ticks.x = element_blank(),
          panel.grid.major.x = element_blank()
          ) +
          labs(x = NULL, y = "PMI, hrs") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
 
ylim(0, max(
                meta.list2$pmi$PMI_hrs)*1.15)
 

pmi.plot



```

```{r fig.width=2, fig.height=3}
ph.plot <-
  ggplot(data = meta.list2$ph, aes(x = 1, y = pH_liquor, dodge = Diagnosis, fill = Diagnosis)) +
  
  
  
    
  
  geom_point(aes(y = pH_liquor, color = Diagnosis), 
             position = position_jitterdodge(dodge.width = 2.3, jitter.width = 0.2), 
               size = 1.1, alpha = 0.5, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3) +
  
  
  
  geom_signif(annotations = signif(wilcox.meta$ph$p.value),
             xmin = 0.8, 
             xmax = 1.2, 
             y_position = max(meta.list2$ph$pH_liquor)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
  
   stat_summary(fun = "mean", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.75), 
                 mapping = aes(group = Diagnosis), fill = "grey",
                 shape = 23
           ) + 
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          axis.ticks.x = element_blank(),
          panel.grid.major.x = element_blank()
          ) +
          labs(x = NULL, y = "pH") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
 
ylim(0, max(
                meta.list2$ph$pH_liquor)*1.15)
 

ph.plot



```

```{r fig.width=2, fig.height=3}
durat.plot <-
  ggplot(data = meta.list2$duration, aes(x = 1, 
          y = Neuroleptic_treatment_duration_years, dodge = Diagnosis, fill = Diagnosis)) +
  
  
  
    
  
  geom_point(aes(y = Neuroleptic_treatment_duration_years, color = Diagnosis), 
             position = position_jitterdodge(dodge.width = 2.3, jitter.width = 0.2), 
               size = 1.1, alpha = 0.5, show.legend = F) +
  

  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 0.3) +
  
  
  
  geom_signif(annotations = signif(wilcox.meta$duration$p.value),
             xmin = 0.8, 
             xmax = 1.2, 
             y_position = max(meta.list2$duration$Neuroleptic_treatment_duration_years)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
   stat_summary(fun = "mean", geom = "point",
                 size = 2,
                 show.legend = F, position = position_dodge(width = 0.75), 
                 mapping = aes(group = Diagnosis), fill = "grey",
                 shape = 23
           ) + 
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
          axis.title.x = element_blank(),
          axis.text.y = element_text(size = 15),
          axis.title.y = element_text(size = 15),
          legend.title = element_blank(),
          legend.text = element_text(size = 15),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 15, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          axis.ticks.x = element_blank(),
          panel.grid.major.x = element_blank()
          ) +
          labs(x = NULL, y = "Neuroleptics, years") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
 
ylim(0, max(
                meta.list2$duration$Neuroleptic_treatment_duration_years)*1.15)
 

durat.plot



```






## FigS Quantitative covariates barplots

```{r fig.width=7, fig.height=3}
covar.plot <-

(age.plot + theme(plot.margin = unit(c(0, 0.8, 0, 0), "cm"))) + 
(pmi.plot + theme(plot.margin = unit(c(0, 0.8, 0, 0), "cm"))) +
(ph.plot + theme(plot.margin = unit(c(0, 0.8, 0, 0), "cm"))) +
(durat.plot + theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))) +
  
  plot_layout(guides = 'collect', ncol = 4) &
  plot_annotation(subtitle = "Numeric covariates") & theme(legend.position = "top", 
                                          plot.subtitle = element_text(size = 15, face = "bold", hjust = 0.7))

covar.plot
```
```{r}
ggsave("./plots/figs1.covar.plot.pdf", covar.plot,
      device = cairo_pdf, width = 7, height = 3, dpi = 600)



```


# IHC Density Plots

## import data
```{r}
cr_dens <- t(read.csv("./cr_densities.csv", header = F, row.names = 1)) %>% as.data.frame %>% 
  `rownames<-`(., NULL) %>% `colnames<-`(., gsub("Layer ", "L", colnames(.)))


pv_dens <- t(read.csv("./pv_densities.csv", header = F, row.names = 1)) %>% as.data.frame %>% 
  `rownames<-`(., NULL) %>% `colnames<-`(., gsub("Layer ", "L", colnames(.)))


cb_dens <- t(read.csv("./cb_densities.csv", header = F, row.names = 1)) %>% as.data.frame %>% 
  `rownames<-`(., NULL) %>% `colnames<-`(., gsub("Layer ", "L", colnames(.)))

nissl_dens <- t(read.csv("./nissl_densities.csv", header = F, row.names = 1)) %>% as.data.frame %>% 
  `rownames<-`(., NULL) %>% `colnames<-`(., gsub("Layer ", "L", colnames(.)))


npy_dens <- t(read.csv("./npy_densities.csv", header = F, row.names = 1)) %>% as.data.frame %>% 
  `rownames<-`(., NULL) %>% `colnames<-`(., gsub("Layer ", "L", colnames(.)))

smi_dens <- t(read.csv("./smi_densities.csv", header = F, row.names = 1)) %>% as.data.frame %>% 
  `rownames<-`(., NULL) %>% `colnames<-`(., gsub("Layer ", "L", colnames(.)))


```


## reshape dataframes


```{r}
#CR
cr_dens <-
    melt(cr_dens, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

cr_dens$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


cr_dens$Density %<>% as.numeric


#PV
pv_dens <-
    melt(pv_dens, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

pv_dens$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


pv_dens$Density %<>% as.numeric


#CB
cb_dens <-
    melt(cb_dens, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

cb_dens$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


cb_dens$Density %<>% as.numeric


#nissl
nissl_dens <-
    melt(nissl_dens, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

nissl_dens$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


nissl_dens$Density %<>% as.numeric


#NPY
npy_dens <-
    melt(npy_dens, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

npy_dens$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


npy_dens$Density %<>% as.numeric



#smi
smi_dens <-
    melt(smi_dens, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

smi_dens$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


smi_dens$Density %<>% as.numeric

```


## cr plot

```{r fig.width=4.25, fig.height=3.2}

cr.plot <-

ggplot(data = cr_dens, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  
 # geom_ellipse(aes(x0 = 2.2, y0 = 4700, a = 0.25, b = 1100, angle = 0), color = "red", fill = NA, lwd = 0.3,
  #             n = 10000) +
  
  
  
    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  geom_signif(annotations = "*",
             xmin = 1.8, 
             xmax = 2.2, 
             y_position = max(cr_dens$Density)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = "n=5254", y = bquote("Density," ~ cells/cm^2), subtitle = "Calretinin+ neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
ylim(0, max(cr_dens$Density)*1.15)


cr.plot

```
```{r}
ggsave("./plots/fig3.cr.plot.pdf", cr.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```


## PV plot

```{r fig.width=4.25, fig.height=3.2}

pv.plot <-

ggplot(data = pv_dens, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

  
  
  
    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  geom_signif(annotations = "*",
             xmin = 1.8, 
             xmax = 2.2, 
             y_position = max(pv_dens$Density)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = "n=6507", y = bquote("Density," ~ cells/cm^2), subtitle = "Parvalbumin+ neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
ylim(-60, max(pv_dens$Density)*1.15)


pv.plot

```



```{r}
ggsave("./plots/fig3.pv.plot.pdf", pv.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```



## CB plot

```{r fig.width=4.25, fig.height=3.2}

cb.plot <-

ggplot(data = cb_dens, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

  
  
  
    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = "n=613", y = bquote("Density," ~ cells/cm^2), subtitle = "Calbindin+ neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)])


cb.plot

```



```{r}
ggsave("./plots/fig3.cb.plot.pdf", cb.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```


## NPY plot

```{r fig.width=4.25, fig.height=3.2}

npy.plot <-

ggplot(data = npy_dens, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

  
  
  
    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = "n=208", y = bquote("Density," ~ cells/cm^2), subtitle = "NPY+ neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)])


npy.plot

```



```{r}
ggsave("./plots/fig3.npy.plot.pdf", npy.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```



## NISSL plot

```{r fig.width=4.25, fig.height=3.2}

nissl.plot <-

ggplot(data = nissl_dens, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

  
  
  
    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  geom_signif(annotations = "*",
             xmin = 3.8, 
             xmax = 4.2, 
             y_position = max(nissl_dens$Density)*1.07,
             textsize = 5, #vjust = 0.5
              ) +
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = "n=45446", y = bquote("Density," ~ cells/cm^2), subtitle = "Nissl stained neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
ylim(0, max(nissl_dens$Density)*1.15)


nissl.plot

```



```{r}
ggsave("./plots/fig3.nissl.plot.pdf", nissl.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```



## SMI plot

```{r fig.width=4.25, fig.height=3.2}

smi.plot <-

ggplot(data = smi_dens, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

  
  
  
    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = "n=41310", y = bquote("Density," ~ cells/cm^2), subtitle = "NEFL/NEFM+ neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)])


smi.plot

```



```{r}
ggsave("./plots/fig3.smi.plot.pdf", smi.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```



# CHRFAM7A expression/proportions

## import data
```{r}
fam_prop <- read.csv("./fam_prop.csv")
fam_expr <- read.csv("./fam_expr.csv")

fam_expr$condition %<>% as.factor


```




## process prop data

```{r}

#average technical replicates

fam_prop %<>% group_by(sample, condition) %>% summarize_at(c(
  "total_CR", "FAM_pos_CR_pos", "CR_pos_only", "prop_FAM_CR"),
  mean)

fam_prop$prop_FAM_CR %<>% as.numeric
fam_prop$condition %<>% as.factor
```

## process expression data

```{r}
fam_expr.median <- group_by(fam_expr, sample, condition) %>% summarize_at("expression", median)
fam_expr.median$expression %<>% as.numeric 


fam_expr.log1p <- fam_expr
fam_expr.log1p$expression %<>% log1p

```




## Normality Shapiro-Wilk FAM proportions

```{r}

fam.prop.list <- split(fam_prop, f = fam_prop$condition)


#shapiro test

shapiro.fam.prop <-
  lapply(fam.prop.list, function(x) shapiro.test(x$prop_FAM_CR))






shapiro.fam.prop

#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
## Normality Shapiro-Wilk FAM express PER CELL

```{r}

fam.expr.list <- split(fam_expr, f = fam_expr$condition)


#shapiro test

shapiro.fam.expr <-
  lapply(fam.expr.list, function(x) shapiro.test(x$expression))






shapiro.fam.expr

#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
non-normal distribution

## Normality Shapiro-Wilk FAM express PER SAMPLE MEDIAN

```{r}

fam.expr.list2 <- split(fam_expr.median, f = fam_expr.median$condition)


#shapiro test

shapiro.fam.expr2 <-
  lapply(fam.expr.list2, function(x) shapiro.test(x$expression))






shapiro.fam.expr2

#p<0.05 - non-normal distribution; p>0.05 - normal distribution
```
normal distr


## Equality of variances (Levene's test) FAM PROPS

```{r}
levene.fam.prop <- leveneTest(fam_prop$prop_FAM_CR ~ fam_prop$condition)
 
 levene.fam.prop
 #p < 0.05 - unequal variances


```
equal variances


## Equality of variances (Levene's test) FAM EXPR PER CELL

```{r}
levene.fam.expr <- leveneTest(fam_expr$expression ~ fam_expr$condition)
 
levene.fam.expr
 #p < 0.05 - unequal variances


```
non-equal variances


## Equality of variances (Levene's test) FAM EXPR MEDIAN

```{r}
levene.fam.expr2 <- leveneTest(fam_expr.median$expression ~ fam_expr.median$condition)
 
levene.fam.expr2
 #p < 0.05 - unequal variances


```

equal variances



## T-test - equal variances  FAM PROPS

```{r}



ttest.fam.prop <- t.test(fam_prop$prop_FAM_CR ~ fam_prop$condition, var.equal = T)


ttest.fam.prop




``` 
non-significant



## Independent 2-group Mann-Whitney U test FAM EXPR PER CELL
```{r}

wilcox.fam.expr <- wilcox.test(fam_expr$expression ~ fam_expr$condition)
wilcox.fam.expr

```






## T-test - equal variances  FAM EXPRESSION PER SAMPLE

```{r}



ttest.fam.expr <- t.test(fam_expr.median$expression ~ fam_expr.median$condition, var.equal = T)


ttest.fam.expr




``` 

non-significant difference




## FIG4 FAM EXPRESSION

```{r fig.width=2.09, fig.height=2.2}
fam.expr.plot <-
  ggplot(data = fam_expr.log1p, aes(x = condition, y = expression, fill = condition)) +
  
  
  rasterise(geom_sina(size = 1.4, alpha = 0.3, aes(color = condition), show.legend = T, scale = F), dpi = 600) +
 
  stat_boxplot(geom ='errorbar',  width = 0.4) +
  
  
  geom_boxplot(notch = F, outlier.shape = NA, alpha = 1, width = 0.13, color = "black", fill = "white", coef = 0
               ) +
  
   geom_signif(annotations = paste0("p=", round(wilcox.fam.expr$p.value, digits = 5)),
             xmin = 1, 
             xmax = 2, 
             y_position = max(fam_expr.log1p$expression)*1.1,
             textsize = 4, vjust = -0.5
              ) +
  
  
#  stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
#                 size = 0.7,
#                 show.legend = F, position = position_dodge(width = 0.7), 
#                 mapping = aes_(), fill = "grey",
#                 shape = 23
#           ) +
  
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          
          axis.title.x = element_text(size = 13),
          axis.text.y = element_text(size = 13),
          axis.title.y = element_text(size = 13),
         # axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          text = element_text(family = "Liberation Sans"),
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
         legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")
         ) +
  labs(x = NULL, y = "log(1+Expression)") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
  
ylim(-0.0001, max(fam_expr.log1p$expression)*1.3)


fam.expr.plot

```


## FIG4 FAM PROPORTION

```{r fig.width=2.09, fig.height=2.2}

fam.prop.plot <-

ggplot(data = fam_prop, aes(x = condition, y = prop_FAM_CR, fill = condition)) +
  

  
  
  
    
  
  geom_point(aes(y = prop_FAM_CR, color = condition), 
               size = 1.5, alpha = 0.5, show.legend = F, position = position_jitter(width = 0.2)) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = F, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
  
  
    theme_bw() +
    theme(axis.text.x=element_blank(),
          axis.ticks.x = element_blank(),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = "CR+ neuron fraction\nexpressing CHRFAM7A") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size = 2))) +
  ylim(0.000, max(fam_prop$prop_FAM_CR)*1.1)


fam.prop.plot

```

```{r fig.width=4.18, fig.height=2.3}
fig.4e.plot <-

(fam.prop.plot + theme(plot.margin = unit(c(0, 0, 0, 0), "cm")))+
  
(fam.expr.plot + theme(plot.margin = unit(c(0, 0, 0, 1), "cm"))) +

  plot_layout(guides = 'collect') &
  plot_annotation(subtitle = "CHRFAM107A in Layer 2") & theme(legend.position = "top", 
                                          plot.subtitle = element_text(size = 13, face = "bold", hjust = 0.75))

fig.4e.plot

```




```{r}
ggsave("./plots/fig4e.pdf", fig.4e.plot,
      device = cairo_pdf, width = 4.18, height = 2.3, dpi = 600)



```

# FIG3 CR CRH VIP PLOTS

## import data

```{r}
cr.p_crh.p_vip.p <- read.csv("./cr.p_crh.p_vip.p.csv")
cr.p_crh.p_vip.n <- read.csv("./cr.p_crh.p_vip.n.csv")
cr.p_crh.n_vip.n <- read.csv("./cr.p_crh.n_vip.n.csv")
```

## reshape dataframes

```{r}
#CR CRH VIP
cr.p_crh.p_vip.p <-
    melt(cr.p_crh.p_vip.p, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

cr.p_crh.p_vip.p$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


cr.p_crh.p_vip.p$Density %<>% as.numeric


#CR CRH
cr.p_crh.p_vip.n <-
    melt(cr.p_crh.p_vip.n, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

cr.p_crh.p_vip.n$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


cr.p_crh.p_vip.n$Density %<>% as.numeric


#CR
cr.p_crh.n_vip.n <-
    melt(cr.p_crh.n_vip.n, id.vars = c("sample", "condition"), measure.vars = 
           c("L6", "L5", "L4", "L3", "L2", "L1"), variable.name = "Layer", value.name = "Density")

cr.p_crh.n_vip.n$Layer %<>% factor(., levels = sort(unique(.) %>% as.vector))


cr.p_crh.n_vip.n$Density %<>% as.numeric

```


## CR CRH VIP plot

```{r fig.width=4.25, fig.height=3.2}

cr_crh_vip.plot <-

ggplot(data = cr.p_crh.p_vip.p, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
#  geom_signif(annotations = "*",
#             xmin = 1.8, 
#             xmax = 2.2, 
#             y_position = max(cr_dens$cr.p_crh.p_vip.p)*1.07,
#             textsize = 5, #vjust = 0.5
#              ) +
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = bquote("Density," ~ cells/cm^2), subtitle = "CR+ CRH+ VIP+ neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
ylim(0, max(cr.p_crh.p_vip.p$Density)*1.15)


cr_crh_vip.plot
```

```{r}
ggsave("./plots/fig3.cr_crh_vip.plot.pdf", cr_crh_vip.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```


## CR CRH plot

```{r fig.width=4.25, fig.height=3.2}

cr_crh.plot <-

ggplot(data = cr.p_crh.p_vip.n, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
#  geom_signif(annotations = "*",
#             xmin = 1.8, 
#             xmax = 2.2, 
#             y_position = max(cr_dens$cr.p_crh.p_vip.p)*1.07,
#             textsize = 5, #vjust = 0.5
#              ) +
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = bquote("Density," ~ cells/cm^2), subtitle = "CR+ CRH+ VIP- neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
ylim(-190, max(cr.p_crh.p_vip.n$Density)*1.15)


cr_crh.plot
```

```{r}
ggsave("./plots/fig3.cr_crh.plot.pdf", cr_crh.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```



## CR plot

```{r fig.width=4.25, fig.height=3.2}

cr2.plot <-

ggplot(data = cr.p_crh.n_vip.n, aes(x = Layer, y = Density, dodge = condition, fill = condition)) +
  

    
  
  geom_point(aes(y = Density, color = condition), position = 
               position_jitterdodge(dodge.width = 0.7, jitter.width = 0.2), 
               size = 1.5, alpha = 0.2, show.legend = F) +
  

  
   stat_summary(fun.data = "mean_sdl", fun.args = list(mult = 1), geom = "pointrange",
                 size = 0.4,
                 show.legend = T, position = position_dodge(width = 0.7), 
                 mapping = aes_(group = ~condition, color = ~condition), fill = "grey",
                 shape = 23
           ) +
  
#  geom_signif(annotations = "*",
#             xmin = 1.8, 
#             xmax = 2.2, 
#             y_position = max(cr_dens$cr.p_crh.p_vip.p)*1.07,
#             textsize = 5, #vjust = 0.5
#              ) +
  
  
    theme_bw() +
    theme(axis.text.x=element_text(angle=90, hjust=1, vjust=0.5, size = 13),
          axis.text.y=element_text(size = 13),
          axis.title.x = element_text(size = 13),
          text = element_text(family = "Liberation Sans", size = 13),
          
          legend.title = element_blank(),
          legend.text = element_text(size = 13),
          legend.position = "top",
          
          plot.subtitle = element_text(size = 13, hjust = 0.5, face = "bold"),
          panel.grid.minor = element_blank(),
          legend.margin =  margin(t = -1, b = -2.5, r = 0, l = 0, unit = "mm")) +
          labs(x = NULL, y = bquote("Density," ~ cells/cm^2), subtitle = "CR+ CRH- VIP- neurons") +
    scale_color_manual(values = palette_45_2[c(8, 15)]) +
    scale_fill_manual(values = palette_45_2[c(8, 15)]) +
  
ylim(-1000, max(cr.p_crh.n_vip.n$Density)*1.15)


cr2.plot
```

```{r}
ggsave("./plots/fig3.crp_crhn_vipn.plot.pdf", cr2.plot,
      device = cairo_pdf, width = 4.25, height = 3.2, dpi = 600)



```





